# Making/Simulating the simple data set {#chapter1}

*Only for exon1 (`B1`) and one housekeeping gene (Hk)* 
So, we are using primer pair `geneB1_2` with primer pair `Hk` for the RT-qPCR. 

Let's call the libraries required and make a simplistic data set to understand how the simulated Ct values are created.

> *What is Ct value?*    
Ans: the PCR cycle in which the amplification signal (because of fluorophore) of a gene/transcript reach a detectable amount. It depends on initial (or starting) amount (or concentration) of a particular gene/transcript. So, the more the starting concentration, earlier it get detected. So, highly abundant gene/transcript will have smaller Ct value compared to other transcripts. For example, let's take two known transcripts: transcript A, and transcript B and their starting concentrations were 100µL and 80µL, respectively. Transcript A will have smaller Ct value compared to transcript B.

```{r}
library(purrr)
library(tibble)

set.seed(10)
df <- data.frame(replicate(n = 3, rnorm(5, 12, .1), simplify = TRUE)) 
df
```
The code above is an example code making the data set with Ct values for 3 replicates (in 3 columns) of the same samples for an exon. We are having total 5 samples (in 5 rows) here. `data.frame()` function makes the array of values into a data frame. I hope you understand how the code is working. Now, let's do a real simulation. 

## Samples of different age

We will have samples of three age groups, namely 15 days, 1 month and 3 months old. In each age group, there will be three genotypes/lines, namely knock-out (KO), tagged (TAG), and wild-type (WT). So, let's add 5 samples per age group. 

### For age 15 days
```{r geneB1}
nKO <- 5 #No. of KO sample
nTAG <- 5 # No. of TAG sample
nWT <- 5 # No. of WT sample

set.seed(11)
geneB1 <- rbind(replicate(n = 3, rnorm(nKO, 25, .1), simplify = TRUE),
               replicate(n = 3, rnorm(nTAG, 22, .08), simplify = TRUE),
               replicate(n = 3, rnorm(nWT, 22, .08), simplify = TRUE)) 
geneB1
```

Have a look at `geneB1` data set. The data set has 3 Ct values in 3 columns for the same sample.

Now, let's make another data set for the housekeeping gene. 

```{r Hk}
Hk <- rbind(replicate(n = 3, rnorm(nKO, 25, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nTAG, 25, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nWT, 25, .1), simplify = TRUE)) #making the dataset with Ct values
Hk
day15 <- data.frame(geneB1, Hk)

colnames(day15) <- c(paste0("geneB1_run", seq(1:3)), paste0("Hk_run", seq(1:3))) #naming the columns. 3 runs
day15$sampleID <- c(paste0("KO", seq(1, nKO)),
                      paste0("TAG", seq(1, nTAG)),
                      paste0("WT", seq(1, nWT))) #naming sample/ID
day15 <- day15[, c(ncol(day15), 1:c(ncol(day15)-1))] #reordering

day15$age <- rep("15days", times = nrow(day15)) #adding age to all. nrow means how many sample
day15$line <- c(rep("KO", times = nKO), rep("TAG", times = nTAG), rep("WT", times = nWT))
day15
```

### For age 1 month
```{r}
geneB1_1 <- rbind(replicate(n = 3, rnorm(nKO, 20, .1), simplify = TRUE),
                          replicate(n = 3, rnorm(nTAG, 17, .07), simplify = TRUE),
                          replicate(n = 3, rnorm(nWT, 17, .07), simplify = TRUE)) 
geneB1_1

Hk1 <- rbind(replicate(n = 3, rnorm(nKO, 25, .1), simplify = TRUE),
               replicate(n = 3, rnorm(nTAG, 25, .1), simplify = TRUE),
               replicate(n = 3, rnorm(nWT, 25, .1), simplify = TRUE)) 
Hk1

month1 <- data.frame(geneB1_1, Hk1)

colnames(month1) <- c(paste0("geneB1_run", seq(1:3)), paste0("Hk_run", seq(1:3))) 
month1$sampleID <- c(paste0("KO", seq(nKO+1, 2*nKO)),
                       paste0("TAG", seq(nTAG+1, 2*nTAG)),
                       paste0("WT", seq(nWT+1, 2*nWT)))

month1 <- month1[, c(ncol(month1), 1:ncol(month1)-1)]

month1$age <- rep("1month", times = nrow(month1))
month1$line <- c(rep("KO", times = nKO), rep("TAG", times = nTAG), rep("WT", times = nWT))
month1
```

### For age 3 months
```{r}
geneB1_2 <- rbind(replicate(n = 3, rnorm(5, 14, .1), simplify = TRUE),
                           replicate(n = 3, rnorm(5, 10, .07), simplify = TRUE),
                           replicate(n = 3, rnorm(5, 10, .07), simplify = TRUE)) 
geneB1_2

Hk2 <- rbind(replicate(n = 3, rnorm(5, 25, .1), simplify = TRUE),
                replicate(n = 3, rnorm(5, 25, .1), simplify = TRUE),
                replicate(n = 3, rnorm(5, 25, .1), simplify = TRUE))
Hk2

month3 <- data.frame(geneB1_2, Hk2)

colnames(month3) <- c(paste0("geneB1_run", seq(1:3)), paste0("Hk_run", seq(1:3))) 
month3$sampleID <- c(paste0("KO", seq(2*nKO+1, 3*nKO)),
                       paste0("TAG", seq(2*nTAG+1, 3*nTAG)),
                       paste0("WT", seq(2*nWT+1, 3*nWT)))
month3 <- month3[, c(ncol(month3), 1:ncol(month3)-1)]

month3$age <- rep("3months", times = nrow(month3)) #adding age to all. nrow means how many sample
month3$line <- c(rep("KO", times = nKO), rep("TAG", times = nTAG), rep("WT", times = nWT))
month3
```

### Making the whole data set combining all three datasets: by `rbind()` function

```{r}
pcr <- rbind(day15, month1, month3)
pcr
```
Our data simulation is done. 

## Data preparation

Now, let's select the numeric columns with Ct values only and find out the mean value for `geneB1` and `HK` from the three runs (or replicates) per sample.

```{r}
a <- pcr %>% select_if(is.numeric) %>% colnames()

pcr$Ct_geneB1 <- apply(pcr[, c(grep("geneB1_", a, ignore.case = TRUE, value = T))], 1, mean)
pcr$Ct_Hk <- apply(pcr[, c(grep("Hk_", a, ignore.case = TRUE, value = T))], 1, mean)
pcr
```

Let's organize the data set. Also, we don't need the raw Ct values from the three runs any more.

```{r}
data <- pcr[, c("sampleID", "age", "line", "Ct_geneB1", "Ct_Hk")]
data
```

Let's apply the formula of ΔCt: **delta_Ct = Ct (gene of interest) – Ct (housekeeping gene)**

```{r}
data$delta_Ct <- data$Ct_geneB1 - data$Ct_Hk 
data
```

Let's calculate the calibrator's mean ΔCt value. Geometric mean is better if calibrators' ΔCt values are variable. Geometric mean is resistant to outlier. Our Ct values are not so variable. So, we can do arithmetic mean.

> calibrator will be the samples relative to which we want to know the expression level. We want to know the fold-change (fold gene expression level) compared to 15-days-old wild-type samples (calibraators) here. We have to keep in mind this thing for our downstream analysis.

```{r}
calibrator <- data[data$line == "WT" & data$age == "15days",]
calibrator
calibrator_deltaCt <- mean(calibrator$delta_Ct)
```

Now, we have to subtract the calibrator ΔCt value from each sample to find out ΔΔCt value

```{r}
data$delta_deltaCt <- data$delta_Ct - calibrator_deltaCt
data
```

Let's find out fold change. The formula for this is: $2^{-ΔΔCt}$

```{r}
data$fold_gene_expression <- 2^-(data$delta_deltaCt)
data
```
> So here, we can see that 15-days-old WT samples have fold change of ~1. Because their Ct values were our calibrator/reference Ct values. Compared to them, 15-days-old TAG samples also have fold change of ~1, which is very reasonable, because we had similar values for the TAG as well. 15-days-old KO samples have less expression (~0.13 fold only). 
But in 1-month-old group, KO samples have fold change of ~4 and TAG/WT samples have fold-change of ~32. So, much higher than the reference group. 
Also, in 3-month-old group, KO samples have fold change of ~250 and WT/TAG samples have fold change of >4020.
So, same pattern (i.e. less in KO and high in WT/TAG) is observed for every `line`. And `age` has an effect in the expression of the gene.

:::{.info}

**Interpreting the fold change**
Have a close look at your fold change column for each line and age group.
:::

To do any statistical analysis, we have to check how the fold change values are distributed. If they are not normally distributed, it is better to log transform them. 

```{r fold-change, fig.width=6, fig.height=6, fig.cap = "Histogram of fold change."}
par(mfrow = c(1,1))
hist(data$fold_gene_expression, xlab="fold change", main="Histogram of fold change")
```
```{r log-fold-change, fig.width=6, fig.height=6, fig.cap = "Histogram of log fold change."}
data$log_fold_change <- log2(data$fold_gene_expression)
par(mfrow = c(1,1))
hist(data$log_fold_change, xlab="log fold change", main="Histogram of log fold change")
```
We can put the same plots together and compare the fold change before and after log transformation.

```{r fold_log-fold-together, fig.width=6, fig.height=3, fig.cap = "Histogram of fold change and log fold change together."}
par(mfrow = c(1,2))
hist(data$fold_gene_expression, xlab="fold change", main="Histogram of fold change")
hist(data$log_fold_change, xlab="log fold change", main="Histogram of log fold change")
```
Our data preparation for one gene of interest and one housekeeping gene is done.   
Hurray......!

Let's save the data set. The code below will save the data set in your working directory. You can change the name as you wish by replacing `pcr1.csv` to the name you want.

```{r}
write.csv(data, "pcr1.csv")
```
> You can work on this data set directly without running the above code if you wnat to do so. In that case, download the data set and read it by naming `data` (by running `data = read.table("pcr1.csv", header = TRUE, sep=",")`. So, all the downstream analysis will be the same.

## Statistical analysis

Let's have a look at the experiment design.
```{r}
table(data$age, data$line)
```
So, it was a balanced design. 

Let's have a look at the log fold change for each age and line combination.
```{r boxplot, fig.width=6, fig.height=6, fig.cap = "Boxplot of log fold change for all age-line combinations."}
boxplot(log_fold_change ~ line*age, data=data, xlab = "Age-Line combinations", ylab = "Log fold change", main="Boxplot of log fold change")
```
The plot shows that WT and TAG in a particular age group have similar expression of transcript B1.

Let's make a column combining age and line together.

```{r}
data$age_line <- paste(data$age, data$line, sep = "_")
```

Let's apply linear model for fold change against `age_line` column.

```{r}
lm1 <- lm(log_fold_change ~ age_line, data=data)
summary(lm1)
```
:::{.info}  
**Interpreting the output:**  
  
R takes the first group as reference by default. So, the comparison is against `15days_KO`, that's why it is not shown in the comparison. It is actually shown in terms of intercept. The value of intercept is an estimate of `15days_KO` (You can have a look at the average of `15days_KO`), which is significant here in this analysis. Here, every group is significantly different than the `15days_KO` group. Have a closer look at your fold change or log fold change columns. Also notice the adjusted R-squared value and overall p-value in the output. Almost 100% variability in the data is captured by our model. It is because we simulated the data this way, there wasn't that much variability.  
:::

**Tips:**  
  
*t value = Estimate/std. Error*. The `t value` measures the size of the difference relative to the variation in the sample data (look at the equation). The greater the magnitude of `t value`, the greater the evidence against the null hypothesis meaning there is greater evidence that there is a significant difference. `t value` closer to 0 means more likely there isn't a significant difference.  
  
**N.B.** you can check the estimate, standard error and t value manually for the reference (`15days_KO`) group.
  

```{r estimate-se, echo=FALSE}
estimate <-  mean(data$log_fold_change[data$age_line == "15days_KO"])
se <- function(x) sqrt(var(x) / length(x))
SE <- se(data$log_fold_change[data$age_line == "15days_KO"]) #`SE` might be a bit different than the value shown in the output table
t_value <- estimate/SE
```

```{r estimate, webex.hide="See the R code to calculate estimate, se and t value", ref.label="estimate-se", eval=FALSE}
```

But we set `15days_WT` as our calibrator/reference group while calculating the ΔCt value. So, let's set `15days_WT` group as our reference. We should relevel the `age_line` column as `factor` for modeling. 

```{r}
data$age_line <- relevel(factor(data$age_line), ref = "15days_WT")
lm2 <- lm(log_fold_change ~ age_line, data=data)
summary(lm2)
```
:::{.info}
**Interpreting the output**  
  
This time, the comparison is relative to `15days_WT`, (and not shown in the comparison). Here every group is significantly different than the `15days_WT` group except `15days_TAG`. And you might have recall why this is the case--we simulated this way, all the WT and TAG samples have basically similar Ct values (or fold change or log fold change). See the boxplot as well. Have a closer look at the values for WT and TAG in the output columns to verify the claim. Also notice the adjusted R-squared value and overall p-value again.  
:::

We can check pair-wise comparisons. It will show all possible combination of pair-wise comparisons. 

```{r}
library(emmeans)
em=emmeans(lm2,~age_line)
emp=emmeans(lm2,pairwise~age_line) 
em
emp
```

Let's have a look from the plotting.
```{r comparison-plot, fig.width=6, fig.height=6, fig.cap = "Comparing all age-line combination"}
plot(em, comparisons = TRUE, ylab = "age-line combination")
```
Have a look at every combination of WT and TAG with age? How are they? Aren't they same as we expected?

What if we model for `age` and `line` separately?

```{r}
lm3 <- lm(log_fold_change ~ age + line, data=data)
summary(lm3)
```
It shows everything very well. But we didn't wanted the comparison/interpretation in this way. It takes all the lines to show things for a age group and all the age groups for a particular line. 

What if we model for interaction of `age` and `line`?

```{r}
lm4 <- lm(log_fold_change ~ age*line, data=data)
summary(lm4)
```
14days old and KO are not shown again. Because R took them as reference, by default. Our `lm2` is better to compare everything.

> The above code is equivalent to `lm(log_fold_change ~ age + line + age*line, data=data)`. We used the short convention here.  
  
:::{.info}  
  
The intercept here is the mean of 15-days-old KO mice samples. Check it by `mean(data$log_fold_change[data$age == "15days" & data$line == "KO"])`. All other results are relative to this group (not shown in the output). You can try `lm(log_fold_change ~ -1 + age*line, data=data)` command as well to see what happens.  
:::

How is our model? Let's check.
```{r}
anova(lm2)
```
The F value is too high. The "F value” of the ANOVA table represents the value of F-statistic $\frac{SSR/1}{SSE/(n-1)}$, where SSR = variation explained by the regression (fitted or model) line and SSE = variation around the conditional mean. The null hypothesis here is no linear dependence of `log-fold-change` on `age_line`. The ANOVA table shows that  the null hypothesis is not true, so `log_fold_change` depends on `age_line` (have a look at the p value). So, our model is very good.  
  
We could use `lme4` or `limma` packages for the modeling as well. But `lm()` function from base R is good enough to do this simple modeling. We will see their use later. Next part of this endeavor will be simulation and analysis of RT-qPCR data with multiple genes of interest/transcripts and multiple reference/housekeeping genes. Stay tuned!
