--- 
title: "RT-qPCR Data Simulation and Analysis"
author: "Md Rasheduzzaman"
date: "2022-08-05"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
# url: your book url like https://bookdown.org/yihui/bookdown
# cover-image: path to the social sharing image like images/cover.jpg
description: |
  This is an example of how to analyze RT-qPCR data.
  The HTML output format for this example is bookdown::gitbook,
  set in the _output.yml file.
link-citations: yes
github-repo: rstudio/bookdown-demo
---

# url: your book url like https://bookdown.org/yihui/bookdown

Placeholder


## How to use the book
## About the experiment design
## Acknowledgement

<!--chapter:end:index.Rmd-->


# Primer efficiency (1 target gene & 1 housekeeping gene)

Placeholder


## Data simulation
## Data analysis

<!--chapter:end:01-intro.Rmd-->


# Making/Simulating the simple data set {#chapter1}

Placeholder


## Samples of different age
### For age 15 days
### For age 1 month
### For age 3 months
### Making the whole data set combining all three datasets: by `rbind()` function
## Data preparation
## Statistical analysis

<!--chapter:end:02-cross-refs.Rmd-->

# Chapter 4
* For multiple gene of interest and multiple housekeeping genes. Before this part, see chapter 1 `\@ref(chapter1)` for better understanding.*

* Let's add two more gene of interest (GOI) or transcripts and one more housekeeping gene

### For age 15 days

Let's add 3 Ct values for all three gene of interest for each sample.

```{r}
library(purrr)
library(tibble)
library(tidyverse)
nKO <- 5 #No. of KO sample
nTAG <- 5 # No. of TAG sample
nWT <- 5 # No. of WT sample

set.seed(11)
geneB1 <- rbind(replicate(n = 3, rnorm(nKO, 20, .1), simplify = TRUE),
               replicate(n = 3, rnorm(nTAG, 19, .08), simplify = TRUE),
               replicate(n = 3, rnorm(nWT, 19, .08), simplify = TRUE)) 
geneB8 <- rbind(replicate(n = 3, rnorm(nKO, 15, .1), simplify = TRUE),
               replicate(n = 3, rnorm(nTAG, 13.5, .08), simplify = TRUE),
               replicate(n = 3, rnorm(nWT, 13.5, .08), simplify = TRUE))
geneB15 <- rbind(replicate(n = 3, rnorm(nKO, 5.5, .1), simplify = TRUE),
               replicate(n = 3, rnorm(nTAG, 5, .08), simplify = TRUE),
               replicate(n = 3, rnorm(nWT, 5, .08), simplify = TRUE))
GOI1 <- data.frame(geneB1, geneB8, geneB15)
```

Let's add 3 Ct values for two housekepping genes for each sample

```{r}
Hk1 <- rbind(replicate(n = 3, rnorm(nKO, 20, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nTAG, 20, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nWT, 20, .1), simplify = TRUE))

Hk2 <- rbind(replicate(n = 3, rnorm(nKO, 22, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nTAG, 22, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nWT, 22, .1), simplify = TRUE))
Hk_1 <- data.frame(Hk1, Hk2)
```

Now, let's add these two data sets.

```{r}
day15 <- data.frame(GOI1, Hk_1)
```

Now, we need to add the column names.

```{r}
colnames(day15) <- c(paste0("geneB1_run", seq(1:3)), paste0("geneB8_run", seq(1:3)), paste0("geneB15_run", seq(1:3)), paste0("Hk1_run", seq(1:3)), paste0("Hk2_run", seq(1:3)))
day15$sampleID <- c(paste0("KO", seq(1, nKO)),
                      paste0("TAG", seq(1, nTAG)),
                      paste0("WT", seq(1, nWT))) #naming sample/ID
day15 <- day15[, c(ncol(day15), 1:ncol(day15) - 1)] #reordering: last column at the beginning and others are in the same sequence
day15$age <- rep("15days", times = nrow(day15)) #adding age to all. nrow means how many sample
day15$line <- c(rep("KO", times = nKO), rep("TAG", times = nTAG), rep("WT", times = nWT))
```

### For age 1 month

Let's add 3 Ct values for all three gene of interest for each sample

```{r}
geneB1_1 <- rbind(replicate(n = 3, rnorm(nKO, 15, .1), simplify = TRUE),
                          replicate(n = 3, rnorm(nTAG, 13, .07), simplify = TRUE),
                          replicate(n = 3, rnorm(nWT, 13, .07), simplify = TRUE))
geneB8_1 <- rbind(replicate(n = 3, rnorm(nKO, 11, .1), simplify = TRUE),
                          replicate(n = 3, rnorm(nTAG, 9.5, .07), simplify = TRUE),
                          replicate(n = 3, rnorm(nWT, 9.5, .07), simplify = TRUE))
geneB15_1 <- rbind(replicate(n = 3, rnorm(nKO, 4, .1), simplify = TRUE),
                          replicate(n = 3, rnorm(nTAG, 3.5, .07), simplify = TRUE),
                          replicate(n = 3, rnorm(nWT, 3.5, .07), simplify = TRUE))
GOI2 <- data.frame(geneB1_1, geneB8_1, geneB15_1)
```

Let's add 3 Ct values for two housekeeping genes for each sample.

```{r}
Hk1_1 <- rbind(replicate(n = 3, rnorm(nKO, 20, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nTAG, 20, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nWT, 20, .1), simplify = TRUE))

Hk2_1 <- rbind(replicate(n = 3, rnorm(nKO, 22, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nTAG, 22, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nWT, 22, .1), simplify = TRUE))
Hk_2 <- data.frame(Hk1_1, Hk2_1)
```

Let's add these two dataset to make the full data set.

```{r}
month1 <- data.frame(GOI2, Hk_2)
```

Now, we need to add the column names.

```{r}
colnames(month1) <- c(paste0("geneB1_run", seq(1:3)), paste0("geneB8_run", seq(1:3)), paste0("geneB15_run", seq(1:3)), paste0("Hk1_run", seq(1:3)), paste0("Hk2_run", seq(1:3)))
month1$sampleID <- c(paste0("KO", seq(nKO+1, 2*nKO)),
                      paste0("TAG", seq(nTAG+1, 2*nTAG)),
                      paste0("WT", seq(nWT+1, 2*nWT)))
month1 <- month1[, c(ncol(month1), 1:ncol(month1) - 1)] 
month1$age <- rep("1month", times = nrow(month1)) 
month1$line <- c(rep("KO", times = nKO), rep("TAG", times = nTAG), rep("WT", times = nWT))
```

### For age 3 months

Let's add 3 Ct values for all three gene of interest for each sample.

```{r}
geneB1_2 <- rbind(replicate(n = 3, rnorm(nKO, 12, .1), simplify = TRUE),
                           replicate(n = 3, rnorm(nTAG, 10, .07), simplify = TRUE),
                           replicate(n = 3, rnorm(nWT, 10, .07), simplify = TRUE))
geneB8_2 <- rbind(replicate(n = 3, rnorm(nKO, 8, .1), simplify = TRUE),
                          replicate(n = 3, rnorm(nTAG, 7, .07), simplify = TRUE),
                          replicate(n = 3, rnorm(nWT, 7, .07), simplify = TRUE))
geneB15_2 <- rbind(replicate(n = 3, rnorm(nKO, 4, .1), simplify = TRUE),
                          replicate(n = 3, rnorm(nTAG, 3, .07), simplify = TRUE),
                          replicate(n = 3, rnorm(nWT, 3, .07), simplify = TRUE))
GOI3 <- data.frame(geneB1_2, geneB8_2, geneB15_2)
```

Let's add 3 Ct values for two housekeeping genes for each sample

```{r}
Hk1_2 <- rbind(replicate(n = 3, rnorm(nKO, 20, .1), simplify = TRUE),
                replicate(n = 3, rnorm(nTAG, 20, .1), simplify = TRUE),
                replicate(n = 3, rnorm(nWT, 20, .1), simplify = TRUE))
Hk2_2 <- rbind(replicate(n = 3, rnorm(nKO, 22, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nTAG, 22, .1), simplify = TRUE),
       replicate(n = 3, rnorm(nWT, 22, .1), simplify = TRUE))
Hk_3 <- data.frame(Hk1_2, Hk2_2)
```

Let's make the full dataset for month3.

```{r}
month3 <- data.frame(GOI3, Hk_3)
```

Naming the columns.

```{r}
colnames(month3) <- c(paste0("geneB1_run", seq(1:3)), paste0("geneB8_run", seq(1:3)), paste0("geneB15_run", seq(1:3)), paste0("Hk1_run", seq(1:3)), paste0("Hk2_run", seq(1:3)))
month3$sampleID <- c(paste0("KO", seq(2*nKO+1, 3*nKO)),
                      paste0("TAG", seq(2*nTAG+1, 3*nTAG)),
                      paste0("WT", seq(2*nWT+1, 3*nWT))) 
month3 <- month3[, c(ncol(month3), 1:ncol(month3) - 1)] 
month3$age <- rep("3months", times = nrow(month3))
month3$line <- c(rep("KO", times = nKO), rep("TAG", times = nTAG), rep("WT", times = nWT))
```

### Making the whole dataset using all three datasets 

We need `rbind()` function for this.

```{r}
pcr1 <- rbind(day15, month1, month3)
```

Now, let's select the numeric columns with Ct values only and find out the mean value for `geneB1`, `geneB8`, `geneB15` and `Hk` from the three runs/replicates per sample

```{r}
a <- pcr1 %>% select_if(is.numeric) %>% colnames()

pcr1$Ct_geneB1 <- apply(pcr1[, c(grep("geneB1_", a, ignore.case = TRUE, value = T))], 1, mean)
pcr1$Ct_geneB8 <- apply(pcr1[, c(grep("geneB8_", a, ignore.case = TRUE, value = T))], 1, mean)
pcr1$Ct_geneB15 <- apply(pcr1[, c(grep("geneB15_", a, ignore.case = TRUE, value = T))], 1, mean)
pcr1$Ct_Hk1 <- apply(pcr1[, c(grep("Hk1", a, ignore.case = TRUE, value = T))], 1, mean)
pcr1$Ct_Hk2 <- apply(pcr1[, c(grep("Hk2", a, ignore.case = TRUE, value = T))], 1, mean)
```

Let's organize the data set. Also, we don't need the raw Ct values from the three runs any more.

```{r}
data1 <- pcr1[, c("sampleID", "age", "line", "Ct_geneB1", "Ct_geneB8", "Ct_geneB15", "Ct_Hk1", "Ct_Hk2")]
```

We need to know primer efficiency for this analysis. The equation for this is: *primer efficiency = (primer efficiency in percentage/100) + 1*. Let's say the efficiencies are like the one in the code chunk (calculated the same way we did in the first chapter).
```{r}
efficiencies <- c(1.98, 2, 2.02, 2.01, 2.01)
names(efficiencies) <- c("B1", "B8", "B15", "Hk1", "Hk2")
```

We need to select a calibrator sample and their average Ct value to get the ∆Ct. We will take WT samples in 15-days-old mice group as our calibrator sample. So, all the expression will be relative to these samples. 

```{r}
calibrator <- data1[data1$line == "WT" & data1$age == "15days",] %>% select_if(is.numeric)
Ct_avg <- apply(calibrator, 2, mean)
```

Let's get the ∆Ct value. ∆Ct = Calibrator Ct - Sample Ct

```{r}
Ct <- data1 %>% select_if(is.numeric)
delta_Ct <- matrix(nrow=nrow(Ct), ncol=ncol(Ct))
for(i in 1:ncol(Ct)) {
  delta_Ct[ , i] <- Ct_avg[[i]] - Ct[ , i] 
}
colnames(delta_Ct) <- c(paste0("delta", names(Ct)))
```

Now we can calculate relative quantity values. The equation is $RQ = E^{\Delta Ct}$. Here, `E` means efficiency.

```{r}
RQ <- matrix(nrow=nrow(delta_Ct), ncol=ncol(delta_Ct))
for(i in 1:length(efficiencies)) {
  RQ[ , i] <-  efficiencies[i]^delta_Ct[,i]
}
colnames(RQ) <- c(paste0("RQ_", names(efficiencies)))
```

Now, we need to calculate the geometric mean of the reference/housekeeping genes RQ values.

```{r}
geoMean_Hk <- apply(RQ[, 4:5], 1, function(x) exp(mean(log(x))))
```

Let's make a dataset with delta_Ct, RQ, geoMean_Hk.

```{r}
final <- cbind(delta_Ct, RQ, geoMean_Hk)
```

Now, we have to calculate relative gene expression values. The equation is: **Relative gene expression = RQ_GOI/GeoMean of housekeeping RQ**

```{r}
nGOI <- 3
Rel_expression <- matrix(nrow=nrow(final), ncol=nGOI)
for(i in 1:nGOI) {
  Rel_expression[ , i] <-  RQ[, i]/final[, "geoMean_Hk"]
}
colnames(Rel_expression) <- c(paste0("RelExp_", names(efficiencies)[1:3]))
final <- data.frame(sampleID=pcr1$sampleID, age=pcr1$age, line=pcr1$line, final, Rel_expression)
```
So, the relative expression of our target transcripts are calculated in the last three columns of the `final` dataset.
Now, we can do statistics on these three columns to know if there is any difference in expression level.

## Statistical analysis

Let's apply statistics

```{r}
stat <- final %>% select(sampleID, age, line, RelExp_B1, RelExp_B8, RelExp_B15)
stat$age_line <- paste(stat$age, stat$line, sep = "_")
lm1 <- lm(RelExp_B1 ~ line + age, data=stat)
lm2 <- lm(RelExp_B1 ~ age_line, data=stat)
summary(lm1)
summary(lm2)

lm2 <- lm(RelExp_B1~ 0 + line * age, data = final1)
summary(lm2)
age <- stat$age
line <- stat$line

df1 <- reshape(stat, direction='long', 
               varying=4:6, 
               timevar='Transcript',
               times=c('B1', 'B8', "B15"),
               v.names=c('Rel_expression_ratio'),
               idvar='sampleID')

df1 <- df1[,c("sampleID", "Transcript", "Rel_expression_ratio", "age", "line")]
df2 <- reshape(df1[,c(1,2,3)], direction = "wide",timevar="sampleID",idvar="Transcript")
df3 <- df2[,-1]
rownames(df3) <- df2[,1]

######################## KO vs TAG #################################
#expression in the first one compared to the second one. 
library(limma)
#design <- model.matrix(~0+line)
#design <- model.matrix(~0+age)
design <- model.matrix(~0+age+line)
fit <- lmFit(df3,design)
#contrast.matrix <- makeContrasts("age3months-age1month",levels=design)
contrast.matrix <- makeContrasts("lineTAG-lineWT",levels=design)
fit <- contrasts.fit(fit,contrast.matrix)
fit=eBayes(fit)
names(fit)
DE <- summary(decideTests(fit,adjust.method="BH",p.value=0.05))
#results = topTable(fit,adjust.method="BH",sort.by="p") 
results = topTable(fit) 
######################## TAG vs WT #################################
fit <- lmFit(df3,design)
contrast.matrix <- makeContrasts("lineTAG-lineWT",levels=design)
fit <- contrasts.fit(fit,contrast.matrix)
fit=eBayes(fit)
names(fit)
DE <- summary(decideTests(fit,adjust.method="BH",p.value=0.05))
results = topTable(fit,adjust.method="BH", sort.by = "p")

####################### KO vs WT ##################################
fit <- lmFit(pcr3,design)
contrast.matrix <- makeContrasts("genotypeKO-genotypeWT",levels=design)
fit <- contrasts.fit(fit,contrast.matrix)
fit=eBayes(fit)
names(fit)
DE <- summary(decideTests(fit,adjust.method="BH",p.value=0.05))
results = topTable(fit,adjust.method="BH", sort.by = "p")

################## 14days vs 1month #############################
#are the genes diff expressed in each age?
design2 <- model.matrix(~0+age)
contrast.matrix <- makeContrasts("age14days-age1month",levels=design2)
fit <- lmFit(pcr3,design2)
fit <- contrasts.fit(fit,contrast.matrix)
fit=eBayes(fit)
names(fit)
DE <- summary(decideTests(fit,adjust.method="BH",p.value=0.05))
results = topTable(fit,adjust.method="BH",sort.by="p")

################## 14days vs 3month ############################
contrast.matrix <- makeContrasts("age14days-age3months",levels=design2)
fit <- lmFit(pcr3,design2)
fit <- contrasts.fit(fit,contrast.matrix)
fit=eBayes(fit)
names(fit)
DE <- summary(decideTests(fit,adjust.method="BH",p.value=0.05))
results = topTable(fit,adjust.method="BH", sort.by = "p")

################## 1month vs 3month ############################
contrast.matrix <- makeContrasts("age1month-age3months",levels=design2)
fit <- lmFit(pcr3,design2)
fit <- contrasts.fit(fit,contrast.matrix)
fit=eBayes(fit)
names(fit)
DE <- summary(decideTests(fit,adjust.method="BH",p.value=0.05))
results = topTable(fit,adjust.method="BH", sort.by = "p")
```

<!--chapter:end:03-multGOI-2Hk.Rmd-->

