<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Chapter 4 | RT-qPCR Data Simulation and Analysis</title>
  <meta name="description" content="<p>This is an example of how to analyze RT-qPCR data.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Chapter 4 | RT-qPCR Data Simulation and Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is an example of how to analyze RT-qPCR data.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Chapter 4 | RT-qPCR Data Simulation and Analysis" />
  
  <meta name="twitter:description" content="<p>This is an example of how to analyze RT-qPCR data.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Md Rasheduzzaman" />


<meta name="date" content="2022-08-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="simulating-the-simple-dataset.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#how-to-use-the-book"><i class="fa fa-check"></i><b>1.1</b> How to use the book</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#about-the-experiment-design"><i class="fa fa-check"></i><b>1.2</b> About the experiment design</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#acknowledgement"><i class="fa fa-check"></i><b>1.3</b> Acknowledgement</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="primer-efficiency-measurement.html"><a href="primer-efficiency-measurement.html"><i class="fa fa-check"></i><b>2</b> Primer Efficiency Measurement</a>
<ul>
<li class="chapter" data-level="2.1" data-path="primer-efficiency-measurement.html"><a href="primer-efficiency-measurement.html#data-simulation"><i class="fa fa-check"></i><b>2.1</b> Data Simulation</a></li>
<li class="chapter" data-level="2.2" data-path="primer-efficiency-measurement.html"><a href="primer-efficiency-measurement.html#data-analysis"><i class="fa fa-check"></i><b>2.2</b> Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="simulating-the-simple-dataset.html"><a href="simulating-the-simple-dataset.html"><i class="fa fa-check"></i><b>3</b> Simulating the Simple Dataset</a>
<ul>
<li class="chapter" data-level="3.1" data-path="simulating-the-simple-dataset.html"><a href="simulating-the-simple-dataset.html#samples-of-different-age"><i class="fa fa-check"></i><b>3.1</b> Samples of different age</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="simulating-the-simple-dataset.html"><a href="simulating-the-simple-dataset.html#for-age-15-days"><i class="fa fa-check"></i><b>3.1.1</b> For age 15 days</a></li>
<li class="chapter" data-level="3.1.2" data-path="simulating-the-simple-dataset.html"><a href="simulating-the-simple-dataset.html#for-age-1-month"><i class="fa fa-check"></i><b>3.1.2</b> For age 1 month</a></li>
<li class="chapter" data-level="3.1.3" data-path="simulating-the-simple-dataset.html"><a href="simulating-the-simple-dataset.html#for-age-3-months"><i class="fa fa-check"></i><b>3.1.3</b> For age 3 months</a></li>
<li class="chapter" data-level="3.1.4" data-path="simulating-the-simple-dataset.html"><a href="simulating-the-simple-dataset.html#making-the-whole-data-set-combining-all-three-datasets-by-rbind-function"><i class="fa fa-check"></i><b>3.1.4</b> Making the whole data set combining all three datasets: by <code>rbind()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="simulating-the-simple-dataset.html"><a href="simulating-the-simple-dataset.html#data-preparation"><i class="fa fa-check"></i><b>3.2</b> Data preparation</a></li>
<li class="chapter" data-level="3.3" data-path="simulating-the-simple-dataset.html"><a href="simulating-the-simple-dataset.html#statistical-analysis"><i class="fa fa-check"></i><b>3.3</b> Statistical analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chapter-4.html"><a href="chapter-4.html"><i class="fa fa-check"></i><b>4</b> Chapter 4</a>
<ul>
<li class="chapter" data-level="4.0.1" data-path="chapter-4.html"><a href="chapter-4.html#for-age-15-days-1"><i class="fa fa-check"></i><b>4.0.1</b> For age 15 days</a></li>
<li class="chapter" data-level="4.0.2" data-path="chapter-4.html"><a href="chapter-4.html#for-age-1-month-1"><i class="fa fa-check"></i><b>4.0.2</b> For age 1 month</a></li>
<li class="chapter" data-level="4.0.3" data-path="chapter-4.html"><a href="chapter-4.html#for-age-3-months-1"><i class="fa fa-check"></i><b>4.0.3</b> For age 3 months</a></li>
<li class="chapter" data-level="4.0.4" data-path="chapter-4.html"><a href="chapter-4.html#making-the-whole-dataset-using-all-three-datasets"><i class="fa fa-check"></i><b>4.0.4</b> Making the whole dataset using all three datasets</a></li>
<li class="chapter" data-level="4.1" data-path="chapter-4.html"><a href="chapter-4.html#statistical-analysis-1"><i class="fa fa-check"></i><b>4.1</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="chapter-4.html"><a href="chapter-4.html#tag-vs-wt"><i class="fa fa-check"></i><b>4.1.1</b> TAG vs WT</a></li>
<li class="chapter" data-level="4.1.2" data-path="chapter-4.html"><a href="chapter-4.html#ko-vs-wt"><i class="fa fa-check"></i><b>4.1.2</b> KO vs WT</a></li>
<li class="chapter" data-level="4.1.3" data-path="chapter-4.html"><a href="chapter-4.html#days-vs-1month"><i class="fa fa-check"></i><b>4.1.3</b> 15days vs 1month</a></li>
<li class="chapter" data-level="4.1.4" data-path="chapter-4.html"><a href="chapter-4.html#days-vs-3month"><i class="fa fa-check"></i><b>4.1.4</b> 15days vs 3month</a></li>
<li class="chapter" data-level="4.1.5" data-path="chapter-4.html"><a href="chapter-4.html#month-vs-3month"><i class="fa fa-check"></i><b>4.1.5</b> 1month vs 3month</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RT-qPCR Data Simulation and Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter-4" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Chapter 4<a href="chapter-4.html#chapter-4" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ul>
<li><p>For multiple gene of interest and multiple housekeeping genes. Before this part, see chapter 1 <code>\@ref(chapter1)</code> for better understanding.*</p></li>
<li><p>Let’s add two more gene of interest (GOI) or transcripts and one more housekeeping gene</p></li>
</ul>
<div id="for-age-15-days-1" class="section level3 hasAnchor" number="4.0.1">
<h3><span class="header-section-number">4.0.1</span> For age 15 days<a href="chapter-4.html#for-age-15-days-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s add 3 Ct values for all three gene of interest for each sample.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="chapter-4.html#cb1-1" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-2"><a href="chapter-4.html#cb1-2" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-3"><a href="chapter-4.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.4     ✔ readr     2.1.4
## ✔ forcats   1.0.0     ✔ stringr   1.5.0
## ✔ ggplot2   3.4.4     ✔ tidyr     1.3.0
## ✔ lubridate 1.9.3     
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="chapter-4.html#cb3-1" tabindex="-1"></a>nKO <span class="ot">&lt;-</span> <span class="dv">5</span> <span class="co">#No. of KO sample</span></span>
<span id="cb3-2"><a href="chapter-4.html#cb3-2" tabindex="-1"></a>nTAG <span class="ot">&lt;-</span> <span class="dv">5</span> <span class="co"># No. of TAG sample</span></span>
<span id="cb3-3"><a href="chapter-4.html#cb3-3" tabindex="-1"></a>nWT <span class="ot">&lt;-</span> <span class="dv">5</span> <span class="co"># No. of WT sample</span></span>
<span id="cb3-4"><a href="chapter-4.html#cb3-4" tabindex="-1"></a></span>
<span id="cb3-5"><a href="chapter-4.html#cb3-5" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">11</span>)</span>
<span id="cb3-6"><a href="chapter-4.html#cb3-6" tabindex="-1"></a>geneB1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-7"><a href="chapter-4.html#cb3-7" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">19</span>, .<span class="dv">08</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-8"><a href="chapter-4.html#cb3-8" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">19</span>, .<span class="dv">08</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb3-9"><a href="chapter-4.html#cb3-9" tabindex="-1"></a>geneB8 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">15</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-10"><a href="chapter-4.html#cb3-10" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="fl">13.5</span>, .<span class="dv">08</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-11"><a href="chapter-4.html#cb3-11" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="fl">13.5</span>, .<span class="dv">08</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-12"><a href="chapter-4.html#cb3-12" tabindex="-1"></a>geneB15 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="fl">5.5</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-13"><a href="chapter-4.html#cb3-13" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">5</span>, .<span class="dv">08</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-14"><a href="chapter-4.html#cb3-14" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">5</span>, .<span class="dv">08</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-15"><a href="chapter-4.html#cb3-15" tabindex="-1"></a>GOI1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(geneB1, geneB8, geneB15)</span></code></pre></div>
<p>Let’s add 3 Ct values for two housekepping genes for each sample</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="chapter-4.html#cb4-1" tabindex="-1"></a>Hk1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-2"><a href="chapter-4.html#cb4-2" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-3"><a href="chapter-4.html#cb4-3" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-4"><a href="chapter-4.html#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="chapter-4.html#cb4-5" tabindex="-1"></a>Hk2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-6"><a href="chapter-4.html#cb4-6" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-7"><a href="chapter-4.html#cb4-7" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-8"><a href="chapter-4.html#cb4-8" tabindex="-1"></a>Hk_1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Hk1, Hk2)</span></code></pre></div>
<p>Now, let’s add these two data sets.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="chapter-4.html#cb5-1" tabindex="-1"></a>day15 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(GOI1, Hk_1)</span></code></pre></div>
<p>Now, we need to add the column names.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="chapter-4.html#cb6-1" tabindex="-1"></a><span class="fu">colnames</span>(day15) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;geneB1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;geneB8_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;geneB15_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk2_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)))</span>
<span id="cb6-2"><a href="chapter-4.html#cb6-2" tabindex="-1"></a>day15<span class="sc">$</span>sampleID <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;KO&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span>, nKO)),</span>
<span id="cb6-3"><a href="chapter-4.html#cb6-3" tabindex="-1"></a>                      <span class="fu">paste0</span>(<span class="st">&quot;TAG&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span>, nTAG)),</span>
<span id="cb6-4"><a href="chapter-4.html#cb6-4" tabindex="-1"></a>                      <span class="fu">paste0</span>(<span class="st">&quot;WT&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span>, nWT))) <span class="co">#naming sample/ID</span></span>
<span id="cb6-5"><a href="chapter-4.html#cb6-5" tabindex="-1"></a>day15 <span class="ot">&lt;-</span> day15[, <span class="fu">c</span>(<span class="fu">ncol</span>(day15), <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(day15) <span class="sc">-</span> <span class="dv">1</span>)] <span class="co">#reordering: last column at the beginning and others are in the same sequence</span></span>
<span id="cb6-6"><a href="chapter-4.html#cb6-6" tabindex="-1"></a>day15<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;15days&quot;</span>, <span class="at">times =</span> <span class="fu">nrow</span>(day15)) <span class="co">#adding age to all. nrow means how many sample</span></span>
<span id="cb6-7"><a href="chapter-4.html#cb6-7" tabindex="-1"></a>day15<span class="sc">$</span>line <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="at">times =</span> nKO), <span class="fu">rep</span>(<span class="st">&quot;TAG&quot;</span>, <span class="at">times =</span> nTAG), <span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="at">times =</span> nWT))</span></code></pre></div>
</div>
<div id="for-age-1-month-1" class="section level3 hasAnchor" number="4.0.2">
<h3><span class="header-section-number">4.0.2</span> For age 1 month<a href="chapter-4.html#for-age-1-month-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s add 3 Ct values for all three gene of interest for each sample</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="chapter-4.html#cb7-1" tabindex="-1"></a>geneB1_1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">15</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb7-2"><a href="chapter-4.html#cb7-2" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">13</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb7-3"><a href="chapter-4.html#cb7-3" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">13</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb7-4"><a href="chapter-4.html#cb7-4" tabindex="-1"></a>geneB8_1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">11</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb7-5"><a href="chapter-4.html#cb7-5" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="fl">9.5</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb7-6"><a href="chapter-4.html#cb7-6" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="fl">9.5</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb7-7"><a href="chapter-4.html#cb7-7" tabindex="-1"></a>geneB15_1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">4</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb7-8"><a href="chapter-4.html#cb7-8" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="fl">3.5</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb7-9"><a href="chapter-4.html#cb7-9" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="fl">3.5</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb7-10"><a href="chapter-4.html#cb7-10" tabindex="-1"></a>GOI2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(geneB1_1, geneB8_1, geneB15_1)</span></code></pre></div>
<p>Let’s add 3 Ct values for two housekeeping genes for each sample.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="chapter-4.html#cb8-1" tabindex="-1"></a>Hk1_1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-2"><a href="chapter-4.html#cb8-2" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-3"><a href="chapter-4.html#cb8-3" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb8-4"><a href="chapter-4.html#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="chapter-4.html#cb8-5" tabindex="-1"></a>Hk2_1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-6"><a href="chapter-4.html#cb8-6" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-7"><a href="chapter-4.html#cb8-7" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb8-8"><a href="chapter-4.html#cb8-8" tabindex="-1"></a>Hk_2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Hk1_1, Hk2_1)</span></code></pre></div>
<p>Let’s add these two dataset to make the full data set.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="chapter-4.html#cb9-1" tabindex="-1"></a>month1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(GOI2, Hk_2)</span></code></pre></div>
<p>Now, we need to add the column names.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="chapter-4.html#cb10-1" tabindex="-1"></a><span class="fu">colnames</span>(month1) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;geneB1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;geneB8_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;geneB15_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk2_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)))</span>
<span id="cb10-2"><a href="chapter-4.html#cb10-2" tabindex="-1"></a>month1<span class="sc">$</span>sampleID <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;KO&quot;</span>, <span class="fu">seq</span>(nKO<span class="sc">+</span><span class="dv">1</span>, <span class="dv">2</span><span class="sc">*</span>nKO)),</span>
<span id="cb10-3"><a href="chapter-4.html#cb10-3" tabindex="-1"></a>                      <span class="fu">paste0</span>(<span class="st">&quot;TAG&quot;</span>, <span class="fu">seq</span>(nTAG<span class="sc">+</span><span class="dv">1</span>, <span class="dv">2</span><span class="sc">*</span>nTAG)),</span>
<span id="cb10-4"><a href="chapter-4.html#cb10-4" tabindex="-1"></a>                      <span class="fu">paste0</span>(<span class="st">&quot;WT&quot;</span>, <span class="fu">seq</span>(nWT<span class="sc">+</span><span class="dv">1</span>, <span class="dv">2</span><span class="sc">*</span>nWT)))</span>
<span id="cb10-5"><a href="chapter-4.html#cb10-5" tabindex="-1"></a>month1 <span class="ot">&lt;-</span> month1[, <span class="fu">c</span>(<span class="fu">ncol</span>(month1), <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(month1) <span class="sc">-</span> <span class="dv">1</span>)] </span>
<span id="cb10-6"><a href="chapter-4.html#cb10-6" tabindex="-1"></a>month1<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;1month&quot;</span>, <span class="at">times =</span> <span class="fu">nrow</span>(month1)) </span>
<span id="cb10-7"><a href="chapter-4.html#cb10-7" tabindex="-1"></a>month1<span class="sc">$</span>line <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="at">times =</span> nKO), <span class="fu">rep</span>(<span class="st">&quot;TAG&quot;</span>, <span class="at">times =</span> nTAG), <span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="at">times =</span> nWT))</span></code></pre></div>
</div>
<div id="for-age-3-months-1" class="section level3 hasAnchor" number="4.0.3">
<h3><span class="header-section-number">4.0.3</span> For age 3 months<a href="chapter-4.html#for-age-3-months-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s add 3 Ct values for all three gene of interest for each sample.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="chapter-4.html#cb11-1" tabindex="-1"></a>geneB1_2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">12</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb11-2"><a href="chapter-4.html#cb11-2" tabindex="-1"></a>                           <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">10</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb11-3"><a href="chapter-4.html#cb11-3" tabindex="-1"></a>                           <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">10</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb11-4"><a href="chapter-4.html#cb11-4" tabindex="-1"></a>geneB8_2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">8</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb11-5"><a href="chapter-4.html#cb11-5" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">7</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb11-6"><a href="chapter-4.html#cb11-6" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">7</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb11-7"><a href="chapter-4.html#cb11-7" tabindex="-1"></a>geneB15_2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">4</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb11-8"><a href="chapter-4.html#cb11-8" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">3</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb11-9"><a href="chapter-4.html#cb11-9" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">3</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb11-10"><a href="chapter-4.html#cb11-10" tabindex="-1"></a>GOI3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(geneB1_2, geneB8_2, geneB15_2)</span></code></pre></div>
<p>Let’s add 3 Ct values for two housekeeping genes for each sample</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="chapter-4.html#cb12-1" tabindex="-1"></a>Hk1_2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb12-2"><a href="chapter-4.html#cb12-2" tabindex="-1"></a>                <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb12-3"><a href="chapter-4.html#cb12-3" tabindex="-1"></a>                <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb12-4"><a href="chapter-4.html#cb12-4" tabindex="-1"></a>Hk2_2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb12-5"><a href="chapter-4.html#cb12-5" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb12-6"><a href="chapter-4.html#cb12-6" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">22</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb12-7"><a href="chapter-4.html#cb12-7" tabindex="-1"></a>Hk_3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Hk1_2, Hk2_2)</span></code></pre></div>
<p>Let’s make the full dataset for month3.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="chapter-4.html#cb13-1" tabindex="-1"></a>month3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(GOI3, Hk_3)</span></code></pre></div>
<p>Naming the columns.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="chapter-4.html#cb14-1" tabindex="-1"></a><span class="fu">colnames</span>(month3) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;geneB1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;geneB8_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;geneB15_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk2_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)))</span>
<span id="cb14-2"><a href="chapter-4.html#cb14-2" tabindex="-1"></a>month3<span class="sc">$</span>sampleID <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;KO&quot;</span>, <span class="fu">seq</span>(<span class="dv">2</span><span class="sc">*</span>nKO<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span><span class="sc">*</span>nKO)),</span>
<span id="cb14-3"><a href="chapter-4.html#cb14-3" tabindex="-1"></a>                      <span class="fu">paste0</span>(<span class="st">&quot;TAG&quot;</span>, <span class="fu">seq</span>(<span class="dv">2</span><span class="sc">*</span>nTAG<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span><span class="sc">*</span>nTAG)),</span>
<span id="cb14-4"><a href="chapter-4.html#cb14-4" tabindex="-1"></a>                      <span class="fu">paste0</span>(<span class="st">&quot;WT&quot;</span>, <span class="fu">seq</span>(<span class="dv">2</span><span class="sc">*</span>nWT<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span><span class="sc">*</span>nWT))) </span>
<span id="cb14-5"><a href="chapter-4.html#cb14-5" tabindex="-1"></a>month3 <span class="ot">&lt;-</span> month3[, <span class="fu">c</span>(<span class="fu">ncol</span>(month3), <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(month3) <span class="sc">-</span> <span class="dv">1</span>)] </span>
<span id="cb14-6"><a href="chapter-4.html#cb14-6" tabindex="-1"></a>month3<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;3months&quot;</span>, <span class="at">times =</span> <span class="fu">nrow</span>(month3))</span>
<span id="cb14-7"><a href="chapter-4.html#cb14-7" tabindex="-1"></a>month3<span class="sc">$</span>line <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="at">times =</span> nKO), <span class="fu">rep</span>(<span class="st">&quot;TAG&quot;</span>, <span class="at">times =</span> nTAG), <span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="at">times =</span> nWT))</span></code></pre></div>
</div>
<div id="making-the-whole-dataset-using-all-three-datasets" class="section level3 hasAnchor" number="4.0.4">
<h3><span class="header-section-number">4.0.4</span> Making the whole dataset using all three datasets<a href="chapter-4.html#making-the-whole-dataset-using-all-three-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We need <code>rbind()</code> function for this.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="chapter-4.html#cb15-1" tabindex="-1"></a>pcr1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(day15, month1, month3)</span></code></pre></div>
<p>Now, let’s select the numeric columns with Ct values only and find out the mean value for <code>geneB1</code>, <code>geneB8</code>, <code>geneB15</code> and <code>Hk</code> from the three runs/replicates per sample</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="chapter-4.html#cb16-1" tabindex="-1"></a>a <span class="ot">&lt;-</span> pcr1 <span class="sc">%&gt;%</span> <span class="fu">select_if</span>(is.numeric) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>()</span>
<span id="cb16-2"><a href="chapter-4.html#cb16-2" tabindex="-1"></a></span>
<span id="cb16-3"><a href="chapter-4.html#cb16-3" tabindex="-1"></a>pcr1<span class="sc">$</span>Ct_geneB1 <span class="ot">&lt;-</span> <span class="fu">apply</span>(pcr1[, <span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&quot;geneB1_&quot;</span>, a, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>, <span class="at">value =</span> T))], <span class="dv">1</span>, mean)</span>
<span id="cb16-4"><a href="chapter-4.html#cb16-4" tabindex="-1"></a>pcr1<span class="sc">$</span>Ct_geneB8 <span class="ot">&lt;-</span> <span class="fu">apply</span>(pcr1[, <span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&quot;geneB8_&quot;</span>, a, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>, <span class="at">value =</span> T))], <span class="dv">1</span>, mean)</span>
<span id="cb16-5"><a href="chapter-4.html#cb16-5" tabindex="-1"></a>pcr1<span class="sc">$</span>Ct_geneB15 <span class="ot">&lt;-</span> <span class="fu">apply</span>(pcr1[, <span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&quot;geneB15_&quot;</span>, a, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>, <span class="at">value =</span> T))], <span class="dv">1</span>, mean)</span>
<span id="cb16-6"><a href="chapter-4.html#cb16-6" tabindex="-1"></a>pcr1<span class="sc">$</span>Ct_Hk1 <span class="ot">&lt;-</span> <span class="fu">apply</span>(pcr1[, <span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&quot;Hk1&quot;</span>, a, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>, <span class="at">value =</span> T))], <span class="dv">1</span>, mean)</span>
<span id="cb16-7"><a href="chapter-4.html#cb16-7" tabindex="-1"></a>pcr1<span class="sc">$</span>Ct_Hk2 <span class="ot">&lt;-</span> <span class="fu">apply</span>(pcr1[, <span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&quot;Hk2&quot;</span>, a, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>, <span class="at">value =</span> T))], <span class="dv">1</span>, mean)</span></code></pre></div>
<p>Let’s organize the data set. Also, we don’t need the raw Ct values from the three runs any more.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="chapter-4.html#cb17-1" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> pcr1[, <span class="fu">c</span>(<span class="st">&quot;sampleID&quot;</span>, <span class="st">&quot;age&quot;</span>, <span class="st">&quot;line&quot;</span>, <span class="st">&quot;Ct_geneB1&quot;</span>, <span class="st">&quot;Ct_geneB8&quot;</span>, <span class="st">&quot;Ct_geneB15&quot;</span>, <span class="st">&quot;Ct_Hk1&quot;</span>, <span class="st">&quot;Ct_Hk2&quot;</span>)]</span></code></pre></div>
<p>We need to know primer efficiency for this analysis. The equation for this is: <em>primer efficiency = (primer efficiency in percentage/100) + 1</em>. Let’s say the efficiencies are like the one in the code chunk (calculated the same way we did in the first chapter).</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="chapter-4.html#cb18-1" tabindex="-1"></a>efficiencies <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.98</span>, <span class="dv">2</span>, <span class="fl">2.02</span>, <span class="fl">2.01</span>, <span class="fl">2.01</span>)</span>
<span id="cb18-2"><a href="chapter-4.html#cb18-2" tabindex="-1"></a><span class="fu">names</span>(efficiencies) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;B1&quot;</span>, <span class="st">&quot;B8&quot;</span>, <span class="st">&quot;B15&quot;</span>, <span class="st">&quot;Hk1&quot;</span>, <span class="st">&quot;Hk2&quot;</span>)</span></code></pre></div>
<p>We need to select a calibrator sample and their average Ct value to get the ∆Ct. We will take WT samples in 15-days-old mice group as our calibrator sample. So, all the expression will be relative to these samples.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="chapter-4.html#cb19-1" tabindex="-1"></a>calibrator <span class="ot">&lt;-</span> data1[data1<span class="sc">$</span>line <span class="sc">==</span> <span class="st">&quot;WT&quot;</span> <span class="sc">&amp;</span> data1<span class="sc">$</span>age <span class="sc">==</span> <span class="st">&quot;15days&quot;</span>,] <span class="sc">%&gt;%</span> <span class="fu">select_if</span>(is.numeric)</span>
<span id="cb19-2"><a href="chapter-4.html#cb19-2" tabindex="-1"></a>Ct_avg <span class="ot">&lt;-</span> <span class="fu">apply</span>(calibrator, <span class="dv">2</span>, mean)</span></code></pre></div>
<p>Let’s get the ∆Ct value. ∆Ct = Calibrator Ct - Sample Ct</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="chapter-4.html#cb20-1" tabindex="-1"></a>Ct <span class="ot">&lt;-</span> data1 <span class="sc">%&gt;%</span> <span class="fu">select_if</span>(is.numeric)</span>
<span id="cb20-2"><a href="chapter-4.html#cb20-2" tabindex="-1"></a>delta_Ct <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">nrow=</span><span class="fu">nrow</span>(Ct), <span class="at">ncol=</span><span class="fu">ncol</span>(Ct))</span>
<span id="cb20-3"><a href="chapter-4.html#cb20-3" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(Ct)) {</span>
<span id="cb20-4"><a href="chapter-4.html#cb20-4" tabindex="-1"></a>  delta_Ct[ , i] <span class="ot">&lt;-</span> Ct_avg[[i]] <span class="sc">-</span> Ct[ , i] </span>
<span id="cb20-5"><a href="chapter-4.html#cb20-5" tabindex="-1"></a>}</span>
<span id="cb20-6"><a href="chapter-4.html#cb20-6" tabindex="-1"></a><span class="fu">colnames</span>(delta_Ct) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;delta&quot;</span>, <span class="fu">names</span>(Ct)))</span></code></pre></div>
<p>Now we can calculate relative quantity values. The equation is <span class="math inline">\(RQ = E^{\Delta Ct}\)</span>. Here, <code>E</code> means efficiency.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="chapter-4.html#cb21-1" tabindex="-1"></a>RQ <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">nrow=</span><span class="fu">nrow</span>(delta_Ct), <span class="at">ncol=</span><span class="fu">ncol</span>(delta_Ct))</span>
<span id="cb21-2"><a href="chapter-4.html#cb21-2" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(efficiencies)) {</span>
<span id="cb21-3"><a href="chapter-4.html#cb21-3" tabindex="-1"></a>  RQ[ , i] <span class="ot">&lt;-</span>  efficiencies[i]<span class="sc">^</span>delta_Ct[,i]</span>
<span id="cb21-4"><a href="chapter-4.html#cb21-4" tabindex="-1"></a>}</span>
<span id="cb21-5"><a href="chapter-4.html#cb21-5" tabindex="-1"></a><span class="fu">colnames</span>(RQ) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;RQ_&quot;</span>, <span class="fu">names</span>(efficiencies)))</span></code></pre></div>
<p>Now, we need to calculate the geometric mean of the reference/housekeeping genes RQ values.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="chapter-4.html#cb22-1" tabindex="-1"></a>geoMean_Hk <span class="ot">&lt;-</span> <span class="fu">apply</span>(RQ[, <span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>], <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">exp</span>(<span class="fu">mean</span>(<span class="fu">log</span>(x))))</span></code></pre></div>
<p>Let’s make a dataset with delta_Ct, RQ, geoMean_Hk.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="chapter-4.html#cb23-1" tabindex="-1"></a>final <span class="ot">&lt;-</span> <span class="fu">cbind</span>(delta_Ct, RQ, geoMean_Hk)</span></code></pre></div>
<p>Now, we have to calculate relative gene expression values. The equation is: <strong>Relative gene expression = RQ_GOI/GeoMean of housekeeping RQ</strong></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="chapter-4.html#cb24-1" tabindex="-1"></a>nGOI <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb24-2"><a href="chapter-4.html#cb24-2" tabindex="-1"></a>Rel_expression <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">nrow=</span><span class="fu">nrow</span>(final), <span class="at">ncol=</span>nGOI)</span>
<span id="cb24-3"><a href="chapter-4.html#cb24-3" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nGOI) {</span>
<span id="cb24-4"><a href="chapter-4.html#cb24-4" tabindex="-1"></a>  Rel_expression[ , i] <span class="ot">&lt;-</span>  RQ[, i]<span class="sc">/</span>final[, <span class="st">&quot;geoMean_Hk&quot;</span>]</span>
<span id="cb24-5"><a href="chapter-4.html#cb24-5" tabindex="-1"></a>}</span>
<span id="cb24-6"><a href="chapter-4.html#cb24-6" tabindex="-1"></a><span class="fu">colnames</span>(Rel_expression) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;RelExp_&quot;</span>, <span class="fu">names</span>(efficiencies)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]))</span>
<span id="cb24-7"><a href="chapter-4.html#cb24-7" tabindex="-1"></a>final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sampleID=</span>pcr1<span class="sc">$</span>sampleID, <span class="at">age=</span>pcr1<span class="sc">$</span>age, <span class="at">line=</span>pcr1<span class="sc">$</span>line, final, Rel_expression)</span></code></pre></div>
<p>So, the relative expression of our target transcripts are calculated in the last three columns of the <code>final</code> dataset.
Now, we can do statistics on these three columns to know if there is any difference in expression level.</p>
</div>
<div id="statistical-analysis-1" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Statistical analysis<a href="chapter-4.html#statistical-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s apply statistics</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="chapter-4.html#cb25-1" tabindex="-1"></a>stat <span class="ot">&lt;-</span> final <span class="sc">%&gt;%</span> <span class="fu">select</span>(sampleID, age, line, RelExp_B1, RelExp_B8, RelExp_B15)</span>
<span id="cb25-2"><a href="chapter-4.html#cb25-2" tabindex="-1"></a>stat<span class="sc">$</span>age_line <span class="ot">&lt;-</span> <span class="fu">paste</span>(stat<span class="sc">$</span>age, stat<span class="sc">$</span>line, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb25-3"><a href="chapter-4.html#cb25-3" tabindex="-1"></a>lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(RelExp_B1 <span class="sc">~</span> line <span class="sc">+</span> age, <span class="at">data=</span>stat)</span>
<span id="cb25-4"><a href="chapter-4.html#cb25-4" tabindex="-1"></a>lm2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(RelExp_B1 <span class="sc">~</span> age_line, <span class="at">data=</span>stat)</span>
<span id="cb25-5"><a href="chapter-4.html#cb25-5" tabindex="-1"></a><span class="fu">summary</span>(lm1)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = RelExp_B1 ~ line + age, data = stat)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -149.84  -42.99  -26.88   64.83   92.33 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)   -85.89      25.68  -3.344   0.0018 ** 
## lineTAG       130.33      28.14   4.632 3.79e-05 ***
## lineWT        129.87      28.14   4.616 3.99e-05 ***
## age1month      44.15      28.14   1.569   0.1245    
## age3months    350.13      28.14  12.444 2.47e-15 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 77.05 on 40 degrees of freedom
## Multiple R-squared:  0.8414, Adjusted R-squared:  0.8256 
## F-statistic: 53.06 on 4 and 40 DF,  p-value: 1.804e-15</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="chapter-4.html#cb27-1" tabindex="-1"></a><span class="fu">summary</span>(lm2)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = RelExp_B1 ~ age_line, data = stat)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -31.0117  -0.8518   0.0121   0.8151  22.0965 
## 
## Coefficients:
##                     Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)           0.5098     3.7326   0.137  0.89213    
## age_line15days_TAG    0.4967     5.2787   0.094  0.92556    
## age_line15days_WT     0.4908     5.2787   0.093  0.92644    
## age_line1month_KO    14.8590     5.2787   2.815  0.00786 ** 
## age_line1month_TAG   58.2152     5.2787  11.028 4.26e-13 ***
## age_line1month_WT    60.3510     5.2787  11.433 1.54e-13 ***
## age_line3months_KO  120.2046     5.2787  22.772  &lt; 2e-16 ***
## age_line3months_TAG 467.3416     5.2787  88.534  &lt; 2e-16 ***
## age_line3months_WT  463.8170     5.2787  87.867  &lt; 2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 8.346 on 36 degrees of freedom
## Multiple R-squared:  0.9983, Adjusted R-squared:  0.998 
## F-statistic:  2683 on 8 and 36 DF,  p-value: &lt; 2.2e-16</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="chapter-4.html#cb29-1" tabindex="-1"></a>lm2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(RelExp_B1<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> line <span class="sc">*</span> age, <span class="at">data =</span> final)</span>
<span id="cb29-2"><a href="chapter-4.html#cb29-2" tabindex="-1"></a><span class="fu">summary</span>(lm2)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = RelExp_B1 ~ 0 + line * age, data = final)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -31.0117  -0.8518   0.0121   0.8151  22.0965 
## 
## Coefficients:
##                    Estimate Std. Error t value Pr(&gt;|t|)    
## lineKO               0.5098     3.7326   0.137  0.89213    
## lineTAG              1.0064     3.7326   0.270  0.78898    
## lineWT               1.0006     3.7326   0.268  0.79018    
## age1month           14.8590     5.2787   2.815  0.00786 ** 
## age3months         120.2046     5.2787  22.772  &lt; 2e-16 ***
## lineTAG:age1month   42.8596     7.4651   5.741 1.54e-06 ***
## lineWT:age1month    45.0013     7.4651   6.028 6.37e-07 ***
## lineTAG:age3months 346.6403     7.4651  46.435  &lt; 2e-16 ***
## lineWT:age3months  343.1216     7.4651  45.963  &lt; 2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 8.346 on 36 degrees of freedom
## Multiple R-squared:  0.9989, Adjusted R-squared:  0.9986 
## F-statistic:  3640 on 9 and 36 DF,  p-value: &lt; 2.2e-16</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="chapter-4.html#cb31-1" tabindex="-1"></a>age <span class="ot">&lt;-</span> stat<span class="sc">$</span>age</span>
<span id="cb31-2"><a href="chapter-4.html#cb31-2" tabindex="-1"></a>line <span class="ot">&lt;-</span> stat<span class="sc">$</span>line</span>
<span id="cb31-3"><a href="chapter-4.html#cb31-3" tabindex="-1"></a></span>
<span id="cb31-4"><a href="chapter-4.html#cb31-4" tabindex="-1"></a>df1 <span class="ot">&lt;-</span> <span class="fu">reshape</span>(stat, <span class="at">direction=</span><span class="st">&#39;long&#39;</span>, </span>
<span id="cb31-5"><a href="chapter-4.html#cb31-5" tabindex="-1"></a>               <span class="at">varying=</span><span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>, </span>
<span id="cb31-6"><a href="chapter-4.html#cb31-6" tabindex="-1"></a>               <span class="at">timevar=</span><span class="st">&#39;Transcript&#39;</span>,</span>
<span id="cb31-7"><a href="chapter-4.html#cb31-7" tabindex="-1"></a>               <span class="at">times=</span><span class="fu">c</span>(<span class="st">&#39;B1&#39;</span>, <span class="st">&#39;B8&#39;</span>, <span class="st">&quot;B15&quot;</span>),</span>
<span id="cb31-8"><a href="chapter-4.html#cb31-8" tabindex="-1"></a>               <span class="at">v.names=</span><span class="fu">c</span>(<span class="st">&#39;Rel_expression_ratio&#39;</span>),</span>
<span id="cb31-9"><a href="chapter-4.html#cb31-9" tabindex="-1"></a>               <span class="at">idvar=</span><span class="st">&#39;sampleID&#39;</span>)</span>
<span id="cb31-10"><a href="chapter-4.html#cb31-10" tabindex="-1"></a></span>
<span id="cb31-11"><a href="chapter-4.html#cb31-11" tabindex="-1"></a>df1 <span class="ot">&lt;-</span> df1[,<span class="fu">c</span>(<span class="st">&quot;sampleID&quot;</span>, <span class="st">&quot;Transcript&quot;</span>, <span class="st">&quot;Rel_expression_ratio&quot;</span>, <span class="st">&quot;age&quot;</span>, <span class="st">&quot;line&quot;</span>)]</span>
<span id="cb31-12"><a href="chapter-4.html#cb31-12" tabindex="-1"></a>df2 <span class="ot">&lt;-</span> <span class="fu">reshape</span>(df1[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)], <span class="at">direction =</span> <span class="st">&quot;wide&quot;</span>,<span class="at">timevar=</span><span class="st">&quot;sampleID&quot;</span>,<span class="at">idvar=</span><span class="st">&quot;Transcript&quot;</span>)</span>
<span id="cb31-13"><a href="chapter-4.html#cb31-13" tabindex="-1"></a>df3 <span class="ot">&lt;-</span> df2[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb31-14"><a href="chapter-4.html#cb31-14" tabindex="-1"></a><span class="fu">rownames</span>(df3) <span class="ot">&lt;-</span> df2[,<span class="dv">1</span>]</span></code></pre></div>
<p>##Comparisons<br />
### KO vs TAG</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="chapter-4.html#cb32-1" tabindex="-1"></a><span class="co">#expression in the first one compared to the second one</span></span>
<span id="cb32-2"><a href="chapter-4.html#cb32-2" tabindex="-1"></a><span class="co">#if (!require(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb32-3"><a href="chapter-4.html#cb32-3" tabindex="-1"></a><span class="co">#    install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb32-4"><a href="chapter-4.html#cb32-4" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb32-5"><a href="chapter-4.html#cb32-5" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;limma&quot;)</span></span>
<span id="cb32-6"><a href="chapter-4.html#cb32-6" tabindex="-1"></a></span>
<span id="cb32-7"><a href="chapter-4.html#cb32-7" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb32-8"><a href="chapter-4.html#cb32-8" tabindex="-1"></a><span class="co">#design &lt;- model.matrix(~0+line)</span></span>
<span id="cb32-9"><a href="chapter-4.html#cb32-9" tabindex="-1"></a><span class="co">#design &lt;- model.matrix(~0+age)</span></span>
<span id="cb32-10"><a href="chapter-4.html#cb32-10" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>line<span class="sc">+</span>age)</span>
<span id="cb32-11"><a href="chapter-4.html#cb32-11" tabindex="-1"></a><span class="co">#design &lt;- model.matrix(~0+age+line)</span></span>
<span id="cb32-12"><a href="chapter-4.html#cb32-12" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(df3,design)</span>
<span id="cb32-13"><a href="chapter-4.html#cb32-13" tabindex="-1"></a><span class="co">#contrast.matrix &lt;- makeContrasts(&quot;age3months-age1month&quot;,levels=design)</span></span>
<span id="cb32-14"><a href="chapter-4.html#cb32-14" tabindex="-1"></a>contrast.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="st">&quot;lineKO-lineTAG&quot;</span>,<span class="at">levels=</span>design)</span>
<span id="cb32-15"><a href="chapter-4.html#cb32-15" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit,contrast.matrix)</span>
<span id="cb32-16"><a href="chapter-4.html#cb32-16" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit)</span>
<span id="cb32-17"><a href="chapter-4.html#cb32-17" tabindex="-1"></a><span class="fu">names</span>(fit)</span></code></pre></div>
<pre><code>##  [1] &quot;coefficients&quot;     &quot;rank&quot;             &quot;assign&quot;           &quot;qr&quot;              
##  [5] &quot;df.residual&quot;      &quot;sigma&quot;            &quot;cov.coefficients&quot; &quot;stdev.unscaled&quot;  
##  [9] &quot;pivot&quot;            &quot;Amean&quot;            &quot;method&quot;           &quot;design&quot;          
## [13] &quot;contrasts&quot;        &quot;df.prior&quot;         &quot;s2.prior&quot;         &quot;var.prior&quot;       
## [17] &quot;proportion&quot;       &quot;s2.post&quot;          &quot;t&quot;                &quot;df.total&quot;        
## [21] &quot;p.value&quot;          &quot;lods&quot;             &quot;F&quot;                &quot;F.p.value&quot;</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="chapter-4.html#cb34-1" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">decideTests</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">p.value=</span><span class="fl">0.05</span>))</span>
<span id="cb34-2"><a href="chapter-4.html#cb34-2" tabindex="-1"></a><span class="co">#results = topTable(fit,adjust.method=&quot;BH&quot;,sort.by=&quot;p&quot;) </span></span>
<span id="cb34-3"><a href="chapter-4.html#cb34-3" tabindex="-1"></a>results <span class="ot">=</span> <span class="fu">topTable</span>(fit) </span></code></pre></div>
<div id="tag-vs-wt" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> TAG vs WT<a href="chapter-4.html#tag-vs-wt" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="chapter-4.html#cb35-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(df3,design)</span>
<span id="cb35-2"><a href="chapter-4.html#cb35-2" tabindex="-1"></a>contrast.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="st">&quot;lineTAG-lineWT&quot;</span>,<span class="at">levels=</span>design)</span>
<span id="cb35-3"><a href="chapter-4.html#cb35-3" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit,contrast.matrix)</span>
<span id="cb35-4"><a href="chapter-4.html#cb35-4" tabindex="-1"></a>fit<span class="ot">=</span><span class="fu">eBayes</span>(fit)</span>
<span id="cb35-5"><a href="chapter-4.html#cb35-5" tabindex="-1"></a><span class="fu">names</span>(fit)</span></code></pre></div>
<pre><code>##  [1] &quot;coefficients&quot;     &quot;rank&quot;             &quot;assign&quot;           &quot;qr&quot;              
##  [5] &quot;df.residual&quot;      &quot;sigma&quot;            &quot;cov.coefficients&quot; &quot;stdev.unscaled&quot;  
##  [9] &quot;pivot&quot;            &quot;Amean&quot;            &quot;method&quot;           &quot;design&quot;          
## [13] &quot;contrasts&quot;        &quot;df.prior&quot;         &quot;s2.prior&quot;         &quot;var.prior&quot;       
## [17] &quot;proportion&quot;       &quot;s2.post&quot;          &quot;t&quot;                &quot;df.total&quot;        
## [21] &quot;p.value&quot;          &quot;lods&quot;             &quot;F&quot;                &quot;F.p.value&quot;</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="chapter-4.html#cb37-1" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">decideTests</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">p.value=</span><span class="fl">0.05</span>))</span>
<span id="cb37-2"><a href="chapter-4.html#cb37-2" tabindex="-1"></a>results <span class="ot">=</span> <span class="fu">topTable</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>, <span class="at">sort.by =</span> <span class="st">&quot;p&quot;</span>)</span></code></pre></div>
</div>
<div id="ko-vs-wt" class="section level3 hasAnchor" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> KO vs WT<a href="chapter-4.html#ko-vs-wt" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="chapter-4.html#cb38-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(df3,design)</span>
<span id="cb38-2"><a href="chapter-4.html#cb38-2" tabindex="-1"></a>contrast.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="st">&quot;lineKO-lineWT&quot;</span>,<span class="at">levels=</span>design)</span>
<span id="cb38-3"><a href="chapter-4.html#cb38-3" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit,contrast.matrix)</span>
<span id="cb38-4"><a href="chapter-4.html#cb38-4" tabindex="-1"></a>fit<span class="ot">=</span><span class="fu">eBayes</span>(fit)</span>
<span id="cb38-5"><a href="chapter-4.html#cb38-5" tabindex="-1"></a><span class="fu">names</span>(fit)</span></code></pre></div>
<pre><code>##  [1] &quot;coefficients&quot;     &quot;rank&quot;             &quot;assign&quot;           &quot;qr&quot;              
##  [5] &quot;df.residual&quot;      &quot;sigma&quot;            &quot;cov.coefficients&quot; &quot;stdev.unscaled&quot;  
##  [9] &quot;pivot&quot;            &quot;Amean&quot;            &quot;method&quot;           &quot;design&quot;          
## [13] &quot;contrasts&quot;        &quot;df.prior&quot;         &quot;s2.prior&quot;         &quot;var.prior&quot;       
## [17] &quot;proportion&quot;       &quot;s2.post&quot;          &quot;t&quot;                &quot;df.total&quot;        
## [21] &quot;p.value&quot;          &quot;lods&quot;             &quot;F&quot;                &quot;F.p.value&quot;</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="chapter-4.html#cb40-1" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">decideTests</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">p.value=</span><span class="fl">0.05</span>))</span>
<span id="cb40-2"><a href="chapter-4.html#cb40-2" tabindex="-1"></a>results <span class="ot">=</span> <span class="fu">topTable</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>, <span class="at">sort.by =</span> <span class="st">&quot;p&quot;</span>)</span></code></pre></div>
</div>
<div id="days-vs-1month" class="section level3 hasAnchor" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> 15days vs 1month<a href="chapter-4.html#days-vs-1month" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="chapter-4.html#cb41-1" tabindex="-1"></a><span class="co">#are the genes diff expressed in each age?</span></span>
<span id="cb41-2"><a href="chapter-4.html#cb41-2" tabindex="-1"></a>design2 <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>age)</span>
<span id="cb41-3"><a href="chapter-4.html#cb41-3" tabindex="-1"></a>contrast.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="st">&quot;age15days-age1month&quot;</span>,<span class="at">levels=</span>design2)</span>
<span id="cb41-4"><a href="chapter-4.html#cb41-4" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(df3,design2)</span>
<span id="cb41-5"><a href="chapter-4.html#cb41-5" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit,contrast.matrix)</span>
<span id="cb41-6"><a href="chapter-4.html#cb41-6" tabindex="-1"></a>fit<span class="ot">=</span><span class="fu">eBayes</span>(fit)</span>
<span id="cb41-7"><a href="chapter-4.html#cb41-7" tabindex="-1"></a><span class="fu">names</span>(fit)</span></code></pre></div>
<pre><code>##  [1] &quot;coefficients&quot;     &quot;rank&quot;             &quot;assign&quot;           &quot;qr&quot;              
##  [5] &quot;df.residual&quot;      &quot;sigma&quot;            &quot;cov.coefficients&quot; &quot;stdev.unscaled&quot;  
##  [9] &quot;pivot&quot;            &quot;Amean&quot;            &quot;method&quot;           &quot;design&quot;          
## [13] &quot;contrasts&quot;        &quot;df.prior&quot;         &quot;s2.prior&quot;         &quot;var.prior&quot;       
## [17] &quot;proportion&quot;       &quot;s2.post&quot;          &quot;t&quot;                &quot;df.total&quot;        
## [21] &quot;p.value&quot;          &quot;lods&quot;             &quot;F&quot;                &quot;F.p.value&quot;</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="chapter-4.html#cb43-1" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">decideTests</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">p.value=</span><span class="fl">0.05</span>))</span>
<span id="cb43-2"><a href="chapter-4.html#cb43-2" tabindex="-1"></a>results <span class="ot">=</span> <span class="fu">topTable</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">sort.by=</span><span class="st">&quot;p&quot;</span>)</span></code></pre></div>
</div>
<div id="days-vs-3month" class="section level3 hasAnchor" number="4.1.4">
<h3><span class="header-section-number">4.1.4</span> 15days vs 3month<a href="chapter-4.html#days-vs-3month" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="chapter-4.html#cb44-1" tabindex="-1"></a>contrast.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="st">&quot;age15days-age3months&quot;</span>,<span class="at">levels=</span>design2)</span>
<span id="cb44-2"><a href="chapter-4.html#cb44-2" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(df3,design2)</span>
<span id="cb44-3"><a href="chapter-4.html#cb44-3" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit,contrast.matrix)</span>
<span id="cb44-4"><a href="chapter-4.html#cb44-4" tabindex="-1"></a>fit<span class="ot">=</span><span class="fu">eBayes</span>(fit)</span>
<span id="cb44-5"><a href="chapter-4.html#cb44-5" tabindex="-1"></a><span class="fu">names</span>(fit)</span></code></pre></div>
<pre><code>##  [1] &quot;coefficients&quot;     &quot;rank&quot;             &quot;assign&quot;           &quot;qr&quot;              
##  [5] &quot;df.residual&quot;      &quot;sigma&quot;            &quot;cov.coefficients&quot; &quot;stdev.unscaled&quot;  
##  [9] &quot;pivot&quot;            &quot;Amean&quot;            &quot;method&quot;           &quot;design&quot;          
## [13] &quot;contrasts&quot;        &quot;df.prior&quot;         &quot;s2.prior&quot;         &quot;var.prior&quot;       
## [17] &quot;proportion&quot;       &quot;s2.post&quot;          &quot;t&quot;                &quot;df.total&quot;        
## [21] &quot;p.value&quot;          &quot;lods&quot;             &quot;F&quot;                &quot;F.p.value&quot;</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="chapter-4.html#cb46-1" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">decideTests</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">p.value=</span><span class="fl">0.05</span>))</span>
<span id="cb46-2"><a href="chapter-4.html#cb46-2" tabindex="-1"></a>results <span class="ot">=</span> <span class="fu">topTable</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>, <span class="at">sort.by =</span> <span class="st">&quot;p&quot;</span>)</span></code></pre></div>
</div>
<div id="month-vs-3month" class="section level3 hasAnchor" number="4.1.5">
<h3><span class="header-section-number">4.1.5</span> 1month vs 3month<a href="chapter-4.html#month-vs-3month" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="chapter-4.html#cb47-1" tabindex="-1"></a>contrast.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="st">&quot;age1month-age3months&quot;</span>,<span class="at">levels=</span>design2)</span>
<span id="cb47-2"><a href="chapter-4.html#cb47-2" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(df3,design2)</span>
<span id="cb47-3"><a href="chapter-4.html#cb47-3" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit,contrast.matrix)</span>
<span id="cb47-4"><a href="chapter-4.html#cb47-4" tabindex="-1"></a>fit<span class="ot">=</span><span class="fu">eBayes</span>(fit)</span>
<span id="cb47-5"><a href="chapter-4.html#cb47-5" tabindex="-1"></a><span class="fu">names</span>(fit)</span></code></pre></div>
<pre><code>##  [1] &quot;coefficients&quot;     &quot;rank&quot;             &quot;assign&quot;           &quot;qr&quot;              
##  [5] &quot;df.residual&quot;      &quot;sigma&quot;            &quot;cov.coefficients&quot; &quot;stdev.unscaled&quot;  
##  [9] &quot;pivot&quot;            &quot;Amean&quot;            &quot;method&quot;           &quot;design&quot;          
## [13] &quot;contrasts&quot;        &quot;df.prior&quot;         &quot;s2.prior&quot;         &quot;var.prior&quot;       
## [17] &quot;proportion&quot;       &quot;s2.post&quot;          &quot;t&quot;                &quot;df.total&quot;        
## [21] &quot;p.value&quot;          &quot;lods&quot;             &quot;F&quot;                &quot;F.p.value&quot;</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="chapter-4.html#cb49-1" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">decideTests</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">p.value=</span><span class="fl">0.05</span>))</span>
<span id="cb49-2"><a href="chapter-4.html#cb49-2" tabindex="-1"></a>results <span class="ot">=</span> <span class="fu">topTable</span>(fit,<span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>, <span class="at">sort.by =</span> <span class="st">&quot;p&quot;</span>)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="simulating-the-simple-dataset.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-multGOI-2Hk.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
