<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Making/Simulating the simple data set (only for exon1 (B1) and one housekeeping gene) | RT-qPCR Data Simulation and Analysis</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Making/Simulating the simple data set (only for exon1 (B1) and one housekeeping gene) | RT-qPCR Data Simulation and Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Making/Simulating the simple data set (only for exon1 (B1) and one housekeeping gene) | RT-qPCR Data Simulation and Analysis" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Md Rasheduzzaman" />


<meta name="date" content="2022-08-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="primer-efficiency-1-target-gene-1-housekeeping-gene.html"/>

<script src="libs/header-attrs-2.14/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> url: your book url like <span>https://bookdown.org/yihui/bookdown</span><span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#how-to-use-the-book"><i class="fa fa-check"></i><b>1.1</b> How to use the book<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#about-the-experiment-design"><i class="fa fa-check"></i><b>1.2</b> About the experiment design<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#render-book"><i class="fa fa-check"></i><b>1.3</b> Render book<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#preview-book"><i class="fa fa-check"></i><b>1.4</b> Preview book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="primer-efficiency-1-target-gene-1-housekeeping-gene.html"><a href="primer-efficiency-1-target-gene-1-housekeeping-gene.html"><i class="fa fa-check"></i><b>2</b> Primer efficiency (1 target gene &amp; 1 housekeeping gene)<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="primer-efficiency-1-target-gene-1-housekeeping-gene.html"><a href="primer-efficiency-1-target-gene-1-housekeeping-gene.html#data-simulation"><i class="fa fa-check"></i><b>2.1</b> Data simulation<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="primer-efficiency-1-target-gene-1-housekeeping-gene.html"><a href="primer-efficiency-1-target-gene-1-housekeeping-gene.html#data-analysis"><i class="fa fa-check"></i><b>2.2</b> Data analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>3</b> Making/Simulating the simple data set <em>(only for exon1 (<code>B1</code>) and one housekeeping gene)</em><span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="chapter1.html"><a href="chapter1.html#samples-of-different-age"><i class="fa fa-check"></i><b>3.1</b> Samples of different age<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="chapter1.html"><a href="chapter1.html#for-age-15-days"><i class="fa fa-check"></i><b>3.1.1</b> For age 15 days<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="chapter1.html"><a href="chapter1.html#for-age-1-month"><i class="fa fa-check"></i><b>3.1.2</b> For age 1 month<span></span></a></li>
<li class="chapter" data-level="3.1.3" data-path="chapter1.html"><a href="chapter1.html#for-age-3-months"><i class="fa fa-check"></i><b>3.1.3</b> For age 3 months<span></span></a></li>
<li class="chapter" data-level="3.1.4" data-path="chapter1.html"><a href="chapter1.html#making-the-whole-data-set-combining-all-three-datasets-by-rbind-function"><i class="fa fa-check"></i><b>3.1.4</b> Making the whole data set combining all three datasets: by <code>rbind()</code> function<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="chapter1.html"><a href="chapter1.html#data-preparation"><i class="fa fa-check"></i><b>3.2</b> Data preparation<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="chapter1.html"><a href="chapter1.html#statistical-analysis"><i class="fa fa-check"></i><b>3.3</b> Statistical analysis<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RT-qPCR Data Simulation and Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter1" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Making/Simulating the simple data set <em>(only for exon1 (<code>B1</code>) and one housekeeping gene)</em><a href="chapter1.html#chapter1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>So, we are using primer pair <code>geneB1_2</code> for the RT-qPCR of geneB1 transcripts.</p>
<p>Let’s call the libraries required and make a simplistic data set to understand how the simulated Ct values are created.</p>
<blockquote>
<p><em>What is Ct value?</em>
Ans: the PCR cycle in which the amplification signal (because of fluorophore) of a gene/transcript reach a detectable amount. It depends on initial (or starting) amount (or concentration) of a particular gene/transcript. So, the more the starting concentration, earlier it get detected. So, highly abundant gene/transcript will have smaller Ct value compared to other transcripts. For example, let’s take two known transcripts: transcript A, and transcript B and their starting concentrations were 100µL and 80µL, respectively. Transcript A will have smaller Ct value compared to transcript B.</p>
</blockquote>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="chapter1.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-2"><a href="chapter1.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-3"><a href="chapter1.html#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="chapter1.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">10</span>)</span>
<span id="cb1-5"><a href="chapter1.html#cb1-5" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(<span class="dv">5</span>, <span class="dv">12</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>)) </span></code></pre></div>
<p>The code above is an example code making the data set with Ct values for 3 replicates (runs) of the same samples for an exon. We are having total 5 samples here. <code>data.frame()</code> function makes the array of values into a data frame. I hope you understand how the code is working. Now, let’s do a real simulation.</p>
<div id="samples-of-different-age" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Samples of different age<a href="chapter1.html#samples-of-different-age" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will have samples of three age groups, namely 15 days, 1 month and 3 months. In each age group, there will be three genotypes/lines, namely knock-out (KO), tagged (TAG), and wild-type (WT). So, let’s add 5 samples per age group.</p>
<div id="for-age-15-days" class="section level3 hasAnchor" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> For age 15 days<a href="chapter1.html#for-age-15-days" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="chapter1.html#cb2-1" aria-hidden="true" tabindex="-1"></a>nKO <span class="ot">&lt;-</span> <span class="dv">5</span> <span class="co">#No. of KO sample</span></span>
<span id="cb2-2"><a href="chapter1.html#cb2-2" aria-hidden="true" tabindex="-1"></a>nTAG <span class="ot">&lt;-</span> <span class="dv">5</span> <span class="co"># No. of TAG sample</span></span>
<span id="cb2-3"><a href="chapter1.html#cb2-3" aria-hidden="true" tabindex="-1"></a>nWT <span class="ot">&lt;-</span> <span class="dv">5</span> <span class="co"># No. of WT sample</span></span>
<span id="cb2-4"><a href="chapter1.html#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="chapter1.html#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">11</span>)</span>
<span id="cb2-6"><a href="chapter1.html#cb2-6" aria-hidden="true" tabindex="-1"></a>geneB1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2-7"><a href="chapter1.html#cb2-7" aria-hidden="true" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">22</span>, .<span class="dv">08</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2-8"><a href="chapter1.html#cb2-8" aria-hidden="true" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">22</span>, .<span class="dv">08</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>)) </span></code></pre></div>
<p>Have a look at <code>geneB1</code> data set. The data set has 3 Ct values in 3 columns for the same sample.</p>
<p>Now, let’s make another data set for the housekeeping gene.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="chapter1.html#cb3-1" aria-hidden="true" tabindex="-1"></a>Hk <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-2"><a href="chapter1.html#cb3-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-3"><a href="chapter1.html#cb3-3" aria-hidden="true" tabindex="-1"></a>       <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>)) <span class="co">#making the dataset with Ct values</span></span>
<span id="cb3-4"><a href="chapter1.html#cb3-4" aria-hidden="true" tabindex="-1"></a>day15 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(geneB1, Hk)</span>
<span id="cb3-5"><a href="chapter1.html#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="chapter1.html#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(day15) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;geneB1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>))) <span class="co">#naming the columns. 3 runs</span></span>
<span id="cb3-7"><a href="chapter1.html#cb3-7" aria-hidden="true" tabindex="-1"></a>day15<span class="sc">$</span>sampleID <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;KO&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span>, nKO)),</span>
<span id="cb3-8"><a href="chapter1.html#cb3-8" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">paste0</span>(<span class="st">&quot;TAG&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span>, nTAG)),</span>
<span id="cb3-9"><a href="chapter1.html#cb3-9" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">paste0</span>(<span class="st">&quot;WT&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span>, nWT))) <span class="co">#naming sample/ID</span></span>
<span id="cb3-10"><a href="chapter1.html#cb3-10" aria-hidden="true" tabindex="-1"></a>day15 <span class="ot">&lt;-</span> day15[, <span class="fu">c</span>(<span class="dv">7</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)] <span class="co">#reordering</span></span>
<span id="cb3-11"><a href="chapter1.html#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="chapter1.html#cb3-12" aria-hidden="true" tabindex="-1"></a>day15<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;15days&quot;</span>, <span class="at">times =</span> <span class="fu">nrow</span>(day15)) <span class="co">#adding age to all. nrow means how many sample</span></span>
<span id="cb3-13"><a href="chapter1.html#cb3-13" aria-hidden="true" tabindex="-1"></a>day15<span class="sc">$</span>line <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="at">times =</span> nKO), <span class="fu">rep</span>(<span class="st">&quot;TAG&quot;</span>, <span class="at">times =</span> nTAG), <span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="at">times =</span> nWT))</span></code></pre></div>
</div>
<div id="for-age-1-month" class="section level3 hasAnchor" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> For age 1 month<a href="chapter1.html#for-age-1-month" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="chapter1.html#cb4-1" aria-hidden="true" tabindex="-1"></a>geneB1_1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">20</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-2"><a href="chapter1.html#cb4-2" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">17</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-3"><a href="chapter1.html#cb4-3" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">17</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb4-4"><a href="chapter1.html#cb4-4" aria-hidden="true" tabindex="-1"></a>Hk1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nKO, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-5"><a href="chapter1.html#cb4-5" aria-hidden="true" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nTAG, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-6"><a href="chapter1.html#cb4-6" aria-hidden="true" tabindex="-1"></a>               <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(nWT, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb4-7"><a href="chapter1.html#cb4-7" aria-hidden="true" tabindex="-1"></a>month1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(geneB1_1, Hk1)</span>
<span id="cb4-8"><a href="chapter1.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#data.frame function makes the array of values into a data frame</span></span>
<span id="cb4-9"><a href="chapter1.html#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="chapter1.html#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(month1) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;geneB1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>))) </span>
<span id="cb4-11"><a href="chapter1.html#cb4-11" aria-hidden="true" tabindex="-1"></a>month1<span class="sc">$</span>sampleID <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;KO&quot;</span>, <span class="fu">seq</span>(nKO<span class="sc">+</span><span class="dv">1</span>, <span class="dv">2</span><span class="sc">*</span>nKO)),</span>
<span id="cb4-12"><a href="chapter1.html#cb4-12" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">paste0</span>(<span class="st">&quot;TAG&quot;</span>, <span class="fu">seq</span>(nTAG<span class="sc">+</span><span class="dv">1</span>, <span class="dv">2</span><span class="sc">*</span>nTAG)),</span>
<span id="cb4-13"><a href="chapter1.html#cb4-13" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">paste0</span>(<span class="st">&quot;WT&quot;</span>, <span class="fu">seq</span>(nWT<span class="sc">+</span><span class="dv">1</span>, <span class="dv">2</span><span class="sc">*</span>nWT)))</span>
<span id="cb4-14"><a href="chapter1.html#cb4-14" aria-hidden="true" tabindex="-1"></a>month1 <span class="ot">&lt;-</span> month1[, <span class="fu">c</span>(<span class="dv">7</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)]</span>
<span id="cb4-15"><a href="chapter1.html#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="chapter1.html#cb4-16" aria-hidden="true" tabindex="-1"></a>month1<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;1month&quot;</span>, <span class="at">times =</span> <span class="fu">nrow</span>(month1))</span>
<span id="cb4-17"><a href="chapter1.html#cb4-17" aria-hidden="true" tabindex="-1"></a>month1<span class="sc">$</span>line <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="at">times =</span> nKO), <span class="fu">rep</span>(<span class="st">&quot;TAG&quot;</span>, <span class="at">times =</span> nTAG), <span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="at">times =</span> nWT))</span></code></pre></div>
</div>
<div id="for-age-3-months" class="section level3 hasAnchor" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> For age 3 months<a href="chapter1.html#for-age-3-months" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="chapter1.html#cb5-1" aria-hidden="true" tabindex="-1"></a>geneB1_2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(<span class="dv">5</span>, <span class="dv">14</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb5-2"><a href="chapter1.html#cb5-2" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(<span class="dv">5</span>, <span class="dv">10</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb5-3"><a href="chapter1.html#cb5-3" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(<span class="dv">5</span>, <span class="dv">10</span>, .<span class="dv">07</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb5-4"><a href="chapter1.html#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="chapter1.html#cb5-5" aria-hidden="true" tabindex="-1"></a>Hk2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(<span class="dv">5</span>, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb5-6"><a href="chapter1.html#cb5-6" aria-hidden="true" tabindex="-1"></a>                <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(<span class="dv">5</span>, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>),</span>
<span id="cb5-7"><a href="chapter1.html#cb5-7" aria-hidden="true" tabindex="-1"></a>                <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">3</span>, <span class="fu">rnorm</span>(<span class="dv">5</span>, <span class="dv">25</span>, .<span class="dv">1</span>), <span class="at">simplify =</span> <span class="cn">TRUE</span>))</span>
<span id="cb5-8"><a href="chapter1.html#cb5-8" aria-hidden="true" tabindex="-1"></a>month3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(geneB1_2, Hk2)</span>
<span id="cb5-9"><a href="chapter1.html#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="chapter1.html#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(month3) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;geneB1_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)), <span class="fu">paste0</span>(<span class="st">&quot;Hk_run&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>))) </span>
<span id="cb5-11"><a href="chapter1.html#cb5-11" aria-hidden="true" tabindex="-1"></a>month3<span class="sc">$</span>sampleID <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;KO&quot;</span>, <span class="fu">seq</span>(<span class="dv">2</span><span class="sc">*</span>nKO<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span><span class="sc">*</span>nKO)),</span>
<span id="cb5-12"><a href="chapter1.html#cb5-12" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">paste0</span>(<span class="st">&quot;TAG&quot;</span>, <span class="fu">seq</span>(<span class="dv">2</span><span class="sc">*</span>nTAG<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span><span class="sc">*</span>nTAG)),</span>
<span id="cb5-13"><a href="chapter1.html#cb5-13" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">paste0</span>(<span class="st">&quot;WT&quot;</span>, <span class="fu">seq</span>(<span class="dv">2</span><span class="sc">*</span>nWT<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span><span class="sc">*</span>nWT)))</span>
<span id="cb5-14"><a href="chapter1.html#cb5-14" aria-hidden="true" tabindex="-1"></a>month3 <span class="ot">&lt;-</span> month3[, <span class="fu">c</span>(<span class="dv">7</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)]</span>
<span id="cb5-15"><a href="chapter1.html#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="chapter1.html#cb5-16" aria-hidden="true" tabindex="-1"></a>month3<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;3months&quot;</span>, <span class="at">times =</span> <span class="fu">nrow</span>(month3)) <span class="co">#adding age to all. nrow means how many sample</span></span>
<span id="cb5-17"><a href="chapter1.html#cb5-17" aria-hidden="true" tabindex="-1"></a>month3<span class="sc">$</span>line <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="at">times =</span> nKO), <span class="fu">rep</span>(<span class="st">&quot;TAG&quot;</span>, <span class="at">times =</span> nTAG), <span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="at">times =</span> nWT))</span></code></pre></div>
</div>
<div id="making-the-whole-data-set-combining-all-three-datasets-by-rbind-function" class="section level3 hasAnchor" number="3.1.4">
<h3><span class="header-section-number">3.1.4</span> Making the whole data set combining all three datasets: by <code>rbind()</code> function<a href="chapter1.html#making-the-whole-data-set-combining-all-three-datasets-by-rbind-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="chapter1.html#cb6-1" aria-hidden="true" tabindex="-1"></a>pcr <span class="ot">&lt;-</span> <span class="fu">rbind</span>(day15, month1, month3)</span></code></pre></div>
<p>Our data simulation is done.</p>
</div>
</div>
<div id="data-preparation" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Data preparation<a href="chapter1.html#data-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now, let’s select the numeric columns with Ct values only and find out the mean value for <code>geneB1</code> and <code>HK</code> from the three runs/replicates per sample.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="chapter1.html#cb7-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> pcr <span class="sc">%&gt;%</span> <span class="fu">select_if</span>(is.numeric) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>()</span>
<span id="cb7-2"><a href="chapter1.html#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="chapter1.html#cb7-3" aria-hidden="true" tabindex="-1"></a>pcr<span class="sc">$</span>Ct_geneB1 <span class="ot">&lt;-</span> <span class="fu">apply</span>(pcr[, <span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&quot;geneB1_&quot;</span>, a, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>, <span class="at">value =</span> T))], <span class="dv">1</span>, mean)</span>
<span id="cb7-4"><a href="chapter1.html#cb7-4" aria-hidden="true" tabindex="-1"></a>pcr<span class="sc">$</span>Ct_Hk <span class="ot">&lt;-</span> <span class="fu">apply</span>(pcr[, <span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&quot;Hk_&quot;</span>, a, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>, <span class="at">value =</span> T))], <span class="dv">1</span>, mean)</span></code></pre></div>
<p>Let’s organize the data set. Also, we don’t need the raw Ct values from the three runs any more.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="chapter1.html#cb8-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> pcr[, <span class="fu">c</span>(<span class="st">&quot;sampleID&quot;</span>, <span class="st">&quot;age&quot;</span>, <span class="st">&quot;line&quot;</span>, <span class="st">&quot;Ct_geneB1&quot;</span>, <span class="st">&quot;Ct_Hk&quot;</span>)]</span></code></pre></div>
<p>Let’s apply the formula of ΔCt: <strong>delta_Ct = Ct (gene of interest) – Ct (housekeeping gene)</strong></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="chapter1.html#cb9-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>delta_Ct <span class="ot">&lt;-</span> data<span class="sc">$</span>Ct_geneB1 <span class="sc">-</span> data<span class="sc">$</span>Ct_Hk </span></code></pre></div>
<p>Let’s calculate the calibrator’s mean ΔCt value. Geometric mean is better if calibrators’ ΔCt values are variable. Geometric mean is resistant to outlier. Our Ct values are not so variable. So, we can do arithmetic mean.</p>
<blockquote>
<p>calibrator will be the samples relative to which we want to know the expression level. We want to know the fold-change (fold gene expression level) compared to 15-days-old wild-type samples (calibraators) here. We have to keep in mind this thing for our downstream analysis.</p>
</blockquote>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="chapter1.html#cb10-1" aria-hidden="true" tabindex="-1"></a>calibrator <span class="ot">&lt;-</span> data[data<span class="sc">$</span>line <span class="sc">==</span> <span class="st">&quot;WT&quot;</span> <span class="sc">&amp;</span> data<span class="sc">$</span>age <span class="sc">==</span> <span class="st">&quot;15days&quot;</span>,]</span>
<span id="cb10-2"><a href="chapter1.html#cb10-2" aria-hidden="true" tabindex="-1"></a>calibrator_deltaCt <span class="ot">&lt;-</span> <span class="fu">mean</span>(calibrator<span class="sc">$</span>delta_Ct)</span></code></pre></div>
<p>Now, we have to subtract the calibrator ΔCt value from each sample to find out ΔΔCt value</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="chapter1.html#cb11-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>delta_deltaCt <span class="ot">&lt;-</span> data<span class="sc">$</span>delta_Ct <span class="sc">-</span> calibrator_deltaCt</span></code></pre></div>
<p>Let’s find out fold change. The formula for this is: <span class="math inline">\(2^{-ΔΔCt}\)</span></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="chapter1.html#cb12-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>fold_gene_expression <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">^-</span>(data<span class="sc">$</span>delta_deltaCt)</span></code></pre></div>
<blockquote>
<p>So here, we can see that 15-days-old WT samples have fold change of ~1. Because their Ct values were our calibrator/reference Ct values. Compared to them, 15-days-old TAG samples also have fold change of ~1, which is very reasonable, because we had similar values for the TAG as well. 15-days-old KO samples have less expression (~0.13 fold only).
But in 1-month-old group, KO samples have fold change of ~4 and TAG/WT samples have fold-change of ~32. So, much higher than the reference group.
Also, in 3-month-old group, KO samples have fold change of ~250 and WT/TAG samples have fold change of &gt;4020.
So, same pattern (i.e. less in KO and high in WT/TAG) is observed for every <code>line</code>. And <code>age</code> has an effect in the expression of the gene.</p>
</blockquote>
<div class="info">
<p><strong>Interpreting the fold change</strong>
Have a close look at your fold change column for each line and age group.</p>
</div>
<p>To do any statistical analysis, we have to check how the fold change values are distributed. If they are not normally distributed, it is better to log transform them.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="chapter1.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb13-2"><a href="chapter1.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(data<span class="sc">$</span>fold_gene_expression, <span class="at">xlab=</span><span class="st">&quot;fold change&quot;</span>, <span class="at">main=</span><span class="st">&quot;Histogram of fold change&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="chapter1.html#cb14-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>log_fold_change <span class="ot">&lt;-</span> <span class="fu">log2</span>(data<span class="sc">$</span>fold_gene_expression)</span>
<span id="cb14-2"><a href="chapter1.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb14-3"><a href="chapter1.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(data<span class="sc">$</span>log_fold_change, <span class="at">xlab=</span><span class="st">&quot;log fold change&quot;</span>, <span class="at">main=</span><span class="st">&quot;Histogram of log fold change&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-13-2.png" width="672" />
We can put the same plots together and compare the fold change before and after log transformation.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="chapter1.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb15-2"><a href="chapter1.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(data<span class="sc">$</span>fold_gene_expression, <span class="at">xlab=</span><span class="st">&quot;fold change&quot;</span>, <span class="at">main=</span><span class="st">&quot;Histogram of fold change&quot;</span>)</span>
<span id="cb15-3"><a href="chapter1.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(data<span class="sc">$</span>log_fold_change, <span class="at">xlab=</span><span class="st">&quot;log fold change&quot;</span>, <span class="at">main=</span><span class="st">&quot;Histogram of log fold change&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-14-1.png" width="672" />
Our data preparation for one gene of interest and one housekeeping gene is done.<br />
Hurray……!</p>
<p>Let’s save the data set. The code below will save the data set in your working directory. You can change the name as you wish by replacing <code>pcr1.csv</code> to the name you want.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="chapter1.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(data, <span class="st">&quot;pcr1.csv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in write.csv(data, &quot;pcr1.csv&quot;, sep = &quot;,&quot;):
## attempt to set &#39;sep&#39; ignored</code></pre>
</div>
<div id="statistical-analysis" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Statistical analysis<a href="chapter1.html#statistical-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s have a look at the experiment design.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="chapter1.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(data<span class="sc">$</span>age, data<span class="sc">$</span>line)</span></code></pre></div>
<pre><code>##          
##           KO TAG WT
##   15days   5   5  5
##   1month   5   5  5
##   3months  5   5  5</code></pre>
<p>So, it was a balanced design.</p>
<p>Let’s have a look at the log fold change for each age and line combination.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="chapter1.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(log_fold_change <span class="sc">~</span> line<span class="sc">*</span>age, <span class="at">data=</span>data, <span class="at">xlab =</span> <span class="st">&quot;Age-Line combinations&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Log fold change&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-17-1.png" width="672" />
The plot shows that WT and TAG in a particular age group have similar expression of transcript B1.</p>
<p>Let’s make a column combining age and line together.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="chapter1.html#cb21-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>age_line <span class="ot">&lt;-</span> <span class="fu">paste</span>(data<span class="sc">$</span>age, data<span class="sc">$</span>line, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span></code></pre></div>
<p>Let’s apply linear model for fold change against <code>age_line</code> column.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="chapter1.html#cb22-1" aria-hidden="true" tabindex="-1"></a>lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(log_fold_change <span class="sc">~</span> age_line, <span class="at">data=</span>data)</span>
<span id="cb22-2"><a href="chapter1.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm1)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = log_fold_change ~ age_line, data = data)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.17576 -0.04204  0.01335  0.04109  0.16291 
## 
## Coefficients:
##                     Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)         -2.93946    0.03387  -86.79   &lt;2e-16
## age_line15days_TAG   2.96739    0.04790   61.95   &lt;2e-16
## age_line15days_WT    2.93946    0.04790   61.37   &lt;2e-16
## age_line1month_KO    4.92432    0.04790  102.81   &lt;2e-16
## age_line1month_TAG   7.93940    0.04790  165.75   &lt;2e-16
## age_line1month_WT    7.93852    0.04790  165.73   &lt;2e-16
## age_line3months_KO  10.90785    0.04790  227.72   &lt;2e-16
## age_line3months_TAG 14.95705    0.04790  312.26   &lt;2e-16
## age_line3months_WT  14.91203    0.04790  311.32   &lt;2e-16
##                        
## (Intercept)         ***
## age_line15days_TAG  ***
## age_line15days_WT   ***
## age_line1month_KO   ***
## age_line1month_TAG  ***
## age_line1month_WT   ***
## age_line3months_KO  ***
## age_line3months_TAG ***
## age_line3months_WT  ***
## ---
## Signif. codes:  
## 0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.07574 on 36 degrees of freedom
## Multiple R-squared:  0.9998, Adjusted R-squared:  0.9998 
## F-statistic: 2.471e+04 on 8 and 36 DF,  p-value: &lt; 2.2e-16</code></pre>
<div class="info">
<p><strong>Interpreting the output:</strong>
R takes the first group as reference by default. So, the comparison is against <code>15days_KO</code>, that’s why it is not shown in the comparison. It is actually shown in terms of intercept. The value of intercept is an estimate of <code>15days_KO</code> (You can have a look at the average of <code>15days_KO</code>), which is significant here in this analysis. Here, every group is significantly different than the <code>15days_KO</code> group. Have a closer look at your fold change or log fold change columns. Also notice the adjusted R-squared value and overall p-value in the output. Almost 100% variability in the data is captured by our model. It is because we simulated the data this way, there wasn’t that much variability.</p>
</div>
<p><strong>Tips:</strong>
<em>t value = Estimate/std. Error</em>. The <code>t value</code> measures the size of the difference relative to the variation in the sample data (look at the equation). The greater the magnitude of <code>t value</code>, the greater the evidence against the null hypothesis meaning there is greater evidence that there is a significant difference. <code>t value</code> closer to 0 means more likely there isn’t a significant difference.</p>
<p><strong>N.B.</strong> you can check the estimate, standard error and t value manually for the reference (<code>15days_KO</code>) group.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="chapter1.html#cb24-1" aria-hidden="true" tabindex="-1"></a>estimate <span class="ot">&lt;-</span>  <span class="fu">mean</span>(data<span class="sc">$</span>log_fold_change[data<span class="sc">$</span>age_line <span class="sc">==</span> <span class="st">&quot;15days_KO&quot;</span>])</span>
<span id="cb24-2"><a href="chapter1.html#cb24-2" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sqrt</span>(<span class="fu">var</span>(x) <span class="sc">/</span> <span class="fu">length</span>(x))</span>
<span id="cb24-3"><a href="chapter1.html#cb24-3" aria-hidden="true" tabindex="-1"></a>SE <span class="ot">&lt;-</span> <span class="fu">se</span>(data<span class="sc">$</span>log_fold_change[data<span class="sc">$</span>age_line <span class="sc">==</span> <span class="st">&quot;15days_KO&quot;</span>]) <span class="co">#`SE` might be a bit different than the value shown in the output table</span></span>
<span id="cb24-4"><a href="chapter1.html#cb24-4" aria-hidden="true" tabindex="-1"></a>t_value <span class="ot">&lt;-</span> estimate<span class="sc">/</span>SE</span></code></pre></div>
<p>But we set <code>15days_WT</code> as our calibrator/reference group while calculating the ΔCt value. So, let’s set <code>15days_WT</code> group as our reference. We should relevel the <code>age_line</code> column as <code>factor</code> as well for modeling.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="chapter1.html#cb25-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>age_line <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(data<span class="sc">$</span>age_line), <span class="at">ref =</span> <span class="st">&quot;15days_WT&quot;</span>)</span>
<span id="cb25-2"><a href="chapter1.html#cb25-2" aria-hidden="true" tabindex="-1"></a>lm2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(log_fold_change <span class="sc">~</span> age_line, <span class="at">data=</span>data)</span>
<span id="cb25-3"><a href="chapter1.html#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm2)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = log_fold_change ~ age_line, data = data)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.17576 -0.04204  0.01335  0.04109  0.16291 
## 
## Coefficients:
##                       Estimate Std. Error t value
## (Intercept)         -1.324e-15  3.387e-02   0.000
## age_line15days_KO   -2.939e+00  4.790e-02 -61.367
## age_line15days_TAG   2.793e-02  4.790e-02   0.583
## age_line1month_KO    1.985e+00  4.790e-02  41.438
## age_line1month_TAG   5.000e+00  4.790e-02 104.384
## age_line1month_WT    4.999e+00  4.790e-02 104.365
## age_line3months_KO   7.968e+00  4.790e-02 166.356
## age_line3months_TAG  1.202e+01  4.790e-02 250.891
## age_line3months_WT   1.197e+01  4.790e-02 249.951
##                     Pr(&gt;|t|)    
## (Intercept)            1.000    
## age_line15days_KO     &lt;2e-16 ***
## age_line15days_TAG     0.563    
## age_line1month_KO     &lt;2e-16 ***
## age_line1month_TAG    &lt;2e-16 ***
## age_line1month_WT     &lt;2e-16 ***
## age_line3months_KO    &lt;2e-16 ***
## age_line3months_TAG   &lt;2e-16 ***
## age_line3months_WT    &lt;2e-16 ***
## ---
## Signif. codes:  
## 0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.07574 on 36 degrees of freedom
## Multiple R-squared:  0.9998, Adjusted R-squared:  0.9998 
## F-statistic: 2.471e+04 on 8 and 36 DF,  p-value: &lt; 2.2e-16</code></pre>
<div class="info">
<p><strong>Interpreting the output</strong>
This time, the comparison is relative to <code>15days_WT</code>, (and not shown in the comparison). Here every group is significantly different than the <code>15days_WT</code> group except <code>15days_TAG</code>. And you might have recall why this is the case–we simulated this way, all the WT and TAG samples have basically similar Ct values (or fold change or log fold change). See the boxplot as well. Have a closer look at the values for WT and TAG in the output columns to verify the claim. Also notice the adjusted R-squared value and overall p-value again.</p>
</div>
<p>We can check pair-wise comparisons. It will show all possible combination of pair-wise comparisons.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="chapter1.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(emmeans)</span>
<span id="cb27-2"><a href="chapter1.html#cb27-2" aria-hidden="true" tabindex="-1"></a>em<span class="ot">=</span><span class="fu">emmeans</span>(lm2,<span class="sc">~</span>age_line)</span>
<span id="cb27-3"><a href="chapter1.html#cb27-3" aria-hidden="true" tabindex="-1"></a>emp<span class="ot">=</span><span class="fu">emmeans</span>(lm2,pairwise<span class="sc">~</span>age_line) </span>
<span id="cb27-4"><a href="chapter1.html#cb27-4" aria-hidden="true" tabindex="-1"></a>em</span></code></pre></div>
<pre><code>##  age_line     emmean     SE df lower.CL upper.CL
##  15days_WT    0.0000 0.0339 36  -0.0687   0.0687
##  15days_KO   -2.9395 0.0339 36  -3.0082  -2.8708
##  15days_TAG   0.0279 0.0339 36  -0.0408   0.0966
##  1month_KO    1.9849 0.0339 36   1.9162   2.0536
##  1month_TAG   4.9999 0.0339 36   4.9313   5.0686
##  1month_WT    4.9991 0.0339 36   4.9304   5.0678
##  3months_KO   7.9684 0.0339 36   7.8997   8.0371
##  3months_TAG 12.0176 0.0339 36  11.9489  12.0863
##  3months_WT  11.9726 0.0339 36  11.9039  12.0413
## 
## Confidence level used: 0.95</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="chapter1.html#cb29-1" aria-hidden="true" tabindex="-1"></a>emp</span></code></pre></div>
<pre><code>## $emmeans
##  age_line     emmean     SE df lower.CL upper.CL
##  15days_WT    0.0000 0.0339 36  -0.0687   0.0687
##  15days_KO   -2.9395 0.0339 36  -3.0082  -2.8708
##  15days_TAG   0.0279 0.0339 36  -0.0408   0.0966
##  1month_KO    1.9849 0.0339 36   1.9162   2.0536
##  1month_TAG   4.9999 0.0339 36   4.9313   5.0686
##  1month_WT    4.9991 0.0339 36   4.9304   5.0678
##  3months_KO   7.9684 0.0339 36   7.8997   8.0371
##  3months_TAG 12.0176 0.0339 36  11.9489  12.0863
##  3months_WT  11.9726 0.0339 36  11.9039  12.0413
## 
## Confidence level used: 0.95 
## 
## $contrasts
##  contrast                  estimate     SE df  t.ratio
##  15days_WT - 15days_KO     2.94e+00 0.0479 36   61.367
##  15days_WT - 15days_TAG   -2.79e-02 0.0479 36   -0.583
##  15days_WT - 1month_KO    -1.98e+00 0.0479 36  -41.438
##  15days_WT - 1month_TAG   -5.00e+00 0.0479 36 -104.384
##  15days_WT - 1month_WT    -5.00e+00 0.0479 36 -104.365
##  15days_WT - 3months_KO   -7.97e+00 0.0479 36 -166.356
##  15days_WT - 3months_TAG  -1.20e+01 0.0479 36 -250.891
##  15days_WT - 3months_WT   -1.20e+01 0.0479 36 -249.951
##  15days_KO - 15days_TAG   -2.97e+00 0.0479 36  -61.950
##  15days_KO - 1month_KO    -4.92e+00 0.0479 36 -102.805
##  15days_KO - 1month_TAG   -7.94e+00 0.0479 36 -165.751
##  15days_KO - 1month_WT    -7.94e+00 0.0479 36 -165.732
##  15days_KO - 3months_KO   -1.09e+01 0.0479 36 -227.723
##  15days_KO - 3months_TAG  -1.50e+01 0.0479 36 -312.258
##  15days_KO - 3months_WT   -1.49e+01 0.0479 36 -311.318
##  15days_TAG - 1month_KO   -1.96e+00 0.0479 36  -40.855
##  15days_TAG - 1month_TAG  -4.97e+00 0.0479 36 -103.801
##  15days_TAG - 1month_WT   -4.97e+00 0.0479 36 -103.782
##  15days_TAG - 3months_KO  -7.94e+00 0.0479 36 -165.773
##  15days_TAG - 3months_TAG -1.20e+01 0.0479 36 -250.308
##  15days_TAG - 3months_WT  -1.19e+01 0.0479 36 -249.368
##  1month_KO - 1month_TAG   -3.02e+00 0.0479 36  -62.946
##  1month_KO - 1month_WT    -3.01e+00 0.0479 36  -62.927
##  1month_KO - 3months_KO   -5.98e+00 0.0479 36 -124.918
##  1month_KO - 3months_TAG  -1.00e+01 0.0479 36 -209.453
##  1month_KO - 3months_WT   -9.99e+00 0.0479 36 -208.513
##  1month_TAG - 1month_WT    8.83e-04 0.0479 36    0.018
##  1month_TAG - 3months_KO  -2.97e+00 0.0479 36  -61.972
##  1month_TAG - 3months_TAG -7.02e+00 0.0479 36 -146.507
##  1month_TAG - 3months_WT  -6.97e+00 0.0479 36 -145.568
##  1month_WT - 3months_KO   -2.97e+00 0.0479 36  -61.991
##  1month_WT - 3months_TAG  -7.02e+00 0.0479 36 -146.526
##  1month_WT - 3months_WT   -6.97e+00 0.0479 36 -145.586
##  3months_KO - 3months_TAG -4.05e+00 0.0479 36  -84.535
##  3months_KO - 3months_WT  -4.00e+00 0.0479 36  -83.595
##  3months_TAG - 3months_WT  4.50e-02 0.0479 36    0.940
##  p.value
##   &lt;.0001
##   0.9996
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   1.0000
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   &lt;.0001
##   0.9888
## 
## P value adjustment: tukey method for comparing a family of 9 estimates</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="chapter1.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(em, <span class="at">comparisons =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-21-1.png" width="672" />
Have a look at every combination of WT and TAG with age? How are they? Aren’t they same as we expected?</p>
<p>What if we model for <code>age</code> and <code>line</code> separately?</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="chapter1.html#cb32-1" aria-hidden="true" tabindex="-1"></a>lm3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(log_fold_change <span class="sc">~</span> age <span class="sc">+</span> line, <span class="at">data=</span>data)</span>
<span id="cb32-2"><a href="chapter1.html#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm3)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = log_fold_change ~ age + line, data = data)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.58563 -0.14537 -0.05045  0.23178  0.40482 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept) -3.19157    0.08549  -37.33   &lt;2e-16 ***
## age1month    4.96513    0.09365   53.02   &lt;2e-16 ***
## age3months  11.62336    0.09365  124.11   &lt;2e-16 ***
## lineTAG      3.34389    0.09365   35.71   &lt;2e-16 ***
## lineWT       3.31928    0.09365   35.44   &lt;2e-16 ***
## ---
## Signif. codes:  
## 0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.2565 on 40 degrees of freedom
## Multiple R-squared:  0.9977, Adjusted R-squared:  0.9974 
## F-statistic:  4300 on 4 and 40 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>It shows everything very well. But we didn’t wanted the comparison/interpretation in this way.</p>
<p>What if we model for interaction of <code>age</code> and <code>line</code>?</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="chapter1.html#cb34-1" aria-hidden="true" tabindex="-1"></a>lm4 <span class="ot">&lt;-</span> <span class="fu">lm</span>(log_fold_change <span class="sc">~</span> age<span class="sc">*</span>line, <span class="at">data=</span>data)</span>
<span id="cb34-2"><a href="chapter1.html#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm4)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = log_fold_change ~ age * line, data = data)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.17576 -0.04204  0.01335  0.04109  0.16291 
## 
## Coefficients:
##                    Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)        -2.93946    0.03387 -86.786   &lt;2e-16
## age1month           4.92432    0.04790 102.805   &lt;2e-16
## age3months         10.90785    0.04790 227.723   &lt;2e-16
## lineTAG             2.96739    0.04790  61.950   &lt;2e-16
## lineWT              2.93946    0.04790  61.367   &lt;2e-16
## age1month:lineTAG   0.04769    0.06774   0.704    0.486
## age3months:lineTAG  1.08181    0.06774  15.970   &lt;2e-16
## age1month:lineWT    0.07474    0.06774   1.103    0.277
## age3months:lineWT   1.06472    0.06774  15.718   &lt;2e-16
##                       
## (Intercept)        ***
## age1month          ***
## age3months         ***
## lineTAG            ***
## lineWT             ***
## age1month:lineTAG     
## age3months:lineTAG ***
## age1month:lineWT      
## age3months:lineWT  ***
## ---
## Signif. codes:  
## 0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.07574 on 36 degrees of freedom
## Multiple R-squared:  0.9998, Adjusted R-squared:  0.9998 
## F-statistic: 2.471e+04 on 8 and 36 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>14days old and KO are not shown again. Because R took them as reference, by default. Our <code>lm2</code> is better to compare everything.</p>
<p>The above code is equivalent to <code>lm(log_fold_change ~ age + line + age*line, data=data)</code>. We used the short convention here.</p>
<div class="info">
<p>The intercept here is the mean of 15-days-old KO mice samples. Check it by <code>mean(data$log_fold_change[data$age == "15days" &amp; data$line == "KO"])</code>. All other results are relative to this group (not shown in the output). You can try <code>lm(log_fold_change ~ -1 + age*line, data=data)</code> command as well to see what happens.</p>
</div>
<p>How is our model? Let’s check.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="chapter1.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(lm2)</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: log_fold_change
##           Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
## age_line   8 1133.86 141.732   24710 &lt; 2.2e-16 ***
## Residuals 36    0.21   0.006                      
## ---
## Signif. codes:  
## 0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>We could use <code>lme4</code> or <code>limma</code> packages for the modeling as well. But <code>lm()</code> function of base R is good enough to do this simple modeling. We will see their use later. Next part of this endeavour will be simulation and analysis of RT-qPCR data with multiple gene of interest/transcripts and multiple reference/housekeeping gene. Stay tuned!</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="primer-efficiency-1-target-gene-1-housekeeping-gene.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-cross-refs.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
